---
title: "Modeling Resources"
date: "August 21, 2023"
date-modified: "`r Sys.Date()`"
format: 
  html:
    page-layout: full
toc: true
toc-location: right
toc-depth: 2
number-sections: true
number-depth: 1
link-external-icon: true
link-external-newwindow: true 
bibliography: references.bib
editor: 
  markdown: 
    wrap: 80
---

```{r echo=FALSE, output=FALSE}

library(webexercises)
```

::: callout-warning
### Disclaimer

This is not a comprehensive list of Protein Modeling resources. Rather, it
contains some of the references, methods, and online servers that I found useful
through the years.

Most useful links (IMHO) are highlighted in bold. Other (many) tools may be
useful for you, feel free to let me know and I'll update the list.
:::

::: callout-note
Links checked on 28/September/2023, let me know if you find any broken link.
:::

# **Template search**

-   **BLAST**: <https://blast.ncbi.nlm.nih.gov/Blast.cgi>

-   **HHPRED**: <https://toolkit.tuebingen.mpg.de/tools/hhpred>

-   **HHblits**: <https://toolkit.tuebingen.mpg.de/tools/hhblits>

-   **JackHMMER**: <https://www.ebi.ac.uk/Tools/hmmer/search/jackhmmer>

    -   Ref. <https://nar.oxfordjournals.org/content/46/W1/W200>

# **Secondary structure and 1D features services**

-   **Jpred4**: <http://www.compbio.dundee.ac.uk/jpred/>

    -   Secondary structure prediction

    -   Ref. <http://nar.oxfordjournals.org/content/early/2015/04/16/nar.gkv332>

-   **PsiPred** : <http://bioinf.cs.ucl.ac.uk/psipred/>

    -   Secondary structure prediction

    -   MEMSAT-SVM (Membrane Helix Prediction)

    -   Ref: <https://academic.oup.com/nar/article/47/W1/W402/5480136>

-   **MULTICOM Toolbox**:
    <http://sysbio.rnet.missouri.edu/multicom_toolbox/web%20services.html>

    -   Pulls together protein structure and structural feature prediction
        tools.

    -   ⚠️Server currently offline. Application available on GitHub:
        <https://github.com/multicom-toolbox>

-   **SCRATCH**: <http://scratch.proteomics.ics.uci.edu/>

    -   Metaserver with a wide range of 1D-3D prediction tools.

-   **SYMPRED**: <http://www.ibi.vu.nl/programs/sympredwww/>

    -   Consensus secondary structure prediction metaserver

-   **MESSA**: <http://prodata.swmed.edu/MESSA/MESSA.cgi>

    -   1D features Metaserver

    -   Ref.<https://bmcbiol.biomedcentral.com/articles/10.1186/1741-7007-10-82>

-   **DeepTMHMM**: <https://dtu.biolib.com/DeepTMHMM>

    -   Deep-learning 2022 version of the classic TMHMM predictor of
        transmembrane helices

    -   Ref. <https://www.biorxiv.org/content/10.1101/2022.04.08.487609v1>

-   **Phobius**: <https://www.ebi.ac.uk/Tools/pfa/phobius/>

    -   Prediction of transmembrane topology and signal peptides from the amino
        acid sequence of a protein.

-   **TopCons**: <http://topcons.cbr.su.se/>

    -   Consensus prediction of membrane protein topology and signal peptides

    -   Ref.
        <https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkv485>

-   **PSORT**: <https://psort.hgc.jp/>

    -   Several localization signals

-   **PRED-TMMB:** <http://bioinformatics.biol.uoa.gr/PRED-TMBB/>

    -   Predicion of beta-barrel TMEs & topology (HMM-based)

-   **PROF-TMF**: <https://open.predictprotein.org/>

    -   Predicion of beta-barrel TMEs (PredictProtein Suite)

-   **NLStradamus**: <http://www.moseslab.csb.utoronto.ca/NLStradamus/>

    -   Prediction of NLS

-   **Sulfinator:** <https://web.expasy.org/sulfinator/>

    -   Prediction of disulfide bonds

-   **DTU Tech resources**: <https://services.healthtech.dtu.dk/>

    -   A collection of molecular biology tools, including a number of services
        for 1D features prediction, such as DeepTMHMM [@hallgren] or NetSurfP
        [@høie2022].\

# **Protein modeling**

-   **SwissModel** [@waterhouse2018]: <https://swissmodel.expasy.org/>

-   PsiPRED [@buchan2019]: <http://bioinf.cs.ucl.ac.uk/psipred/>

    -   Fold recognition server (among other tools)

-   **RaptorX** [@wang2017]: <http://raptorx.uchicago.edu/ContactMap/>

    -   Distance-based Protein Folding Powered by Deep Learning

-   **Phyre2** [@kelley2015]:
    <http://www.sbg.bio.ic.ac.uk/~phyre2/html/page.cgi?id=index>

    -   Fold recognition & fragment assembler

-   **Robetta** server: <https://robetta.bakerlab.org/>

    -   Several methods, including RosettaCM [@song2013] and RoseTTAFOLD
        [@baek2021a; @baeka]

-   **ModWeb** [@pieper2014]**:** <https://modbase.compbio.ucsf.edu/modweb/>

    -   Webserver using MODELLER (free registration required)

-   **ColabFold** [@mirdita2022]: <https://github.com/sokrypton/ColabFold>

    -   Several notebooks available

-   **DeepMind's AlphaFold colab**:
    <https://colab.research.google.com/github/deepmind/alphafold/blob/main/notebooks/AlphaFold.ipynb>

-   **Openfold** [@ahdritz] Colab:
    <https://colab.research.google.com/github/aqlaboratory/openfold/blob/main/notebooks/OpenFold.ipyn>

-   **Unifold** [@li] Colab:
    <https://colab.research.google.com/github/dptech-corp/Uni-Fold/blob/main/notebooks/unifold.ipynb>

-   **ESM Metagenomic Atlas** (DDBB & tools, @lin2023):
    <https://esmatlas.com/about>

-   **PymolFold**: A handy Pymol script that calls the ESM-Fold API

    -   <https://github.com/JinyuanSun/PymolFold>

    -   Web version: <http://103.79.77.89:8501/>

-   **Zhang server**: <https://zhanggroup.org/services/>

    -   Multiple methods, including I-TASSER [@roy2010], C-I-TASSER [@zheng2021]
        or I-TASSER-MTD, among others.

# **Model quality assessment**

-   **QMEAN & QMEANDisCo** [@studer2020]: <https://swissmodel.expasy.org/qmean/>

-   **Modfold** [@mcguffin2021]**:** <https://www.reading.ac.uk/bioinf/ModFOLD/>

-   **VoroMQA** [@olechnovic2019]**:
    <http://bioinformatics.ibt.lt/wtsam/voromqa>**

-   **DeepUMQA** [@guo2022]: <http://zhanglab-bioinf.com/DeepUMQA/>

# **Other resources**

-   **DALI** [@holm2022]: <http://ekhidna2.biocenter.helsinki.fi/dali/>

-   **FoldSeek** [@vankempen2023]: <https://search.foldseek.com/search>

    -   Fast search for structural similarities

-   FatCat [@li2020]**:** <http://fatcat.godziklab.org/fatcat/fatcat_pair.html>

    -   Structural alignment tool.

-   LGA: <http://proteinmodel.org/AS2TS/LGA/lga.html>

    -   LGA program is being developed for structure comparative analysis of two
        selected 3D protein structures or fragments of 3D protein structures

    -   Ref.
        <https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkg571>

-   Galaxy Refine: <http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=REFINE>

    -   A server for automatically refining protein model structures.

    -   Ref. <http://www.ncbi.nlm.nih.gov/pubmed/23737448>

-   Galaxy Refine2:
    [[galaxy.seoklab.org/cgi-bin/submit.cgi?type=REFINE2]{.underline}](http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=REFINE2)

    -   \<300 aa

    -   Ref:<https://academic.oup.com/nar/article/47/W1/W451/5475172>

-   HOMCOS: <http://homcos.pdbj.org/>

    -   Modeling of 3D Structures of Complexes

    -   Ref.
        <http://link.springer.com/article/10.1007/s10969-016-9208-y?wt_mc=Internal.Event.1.SEM.ArticleAuthorOnlineFirst>

-   pyDockWEB: <https://life.bsc.es/pid/pydockweb>

    -   Prediction of protein-protein interactions (hetero-oligomers modeling)

-   Hex: <http://hexserver.loria.fr/>

    -   Prediction of protein-protein interactions (homo- & hetero-oligomers
        modeling)

    -   Ref. <https://onlinelibrary.wiley.com/doi/full/10.1002/prot.24433>

-   DaReUs-Loop**:**
    <https://bioserv.rpbs.univ-paris-diderot.fr/services/DaReUS-Loop/>

    -   Modeling loops

    -   Ref. <https://www.nature.com/articles/s41598-018-32079-w>

-   GalaxyLoop: <http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=LOOP>

    -   Loops refining

    -   Ref.
        <https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0113811>

-   Expasy proteomics tools: <https://www.expasy.org/proteomics>

    -   List of qMultiple links and resources

-   Protein modeling links \@Sali Lab:
    <https://salilab.org/bioinformatics_resources.html>

-   Homology modeling \@Proteopedia:

    -   <https://proteopedia.org/wiki/index.php/Homology_model>

    -   <https://proteopedia.org/wiki/index.php/User:Wayne_Decatur/Homology_Modeling>

-   List of protein structure prediction software \@Wikipedia:
    <https://en.wikipedia.org/wiki/List_of_protein_structure_prediction_software>
