<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.56">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-08-21">

<title>Protein Databases – Structural Bioinformatics at UAM</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./pdb.html" rel="next">
<link href="./intro.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/iconify-1.0.0-beta.2/iconify-icon.min.js"></script>
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="styles.css">
<link rel="stylesheet" href="include/webex.css">
<meta name="twitter:title" content="Protein Databases – Structural Bioinformatics at UAM">
<meta name="twitter:description" content="">
<meta name="twitter:card" content="summary">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./intro.html">Background</a></li><li class="breadcrumb-item"><a href="./ddbb.html">Protein Databases</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Structural Bioinformatics at UAM</a> 
        <div class="sidebar-tools-main">
    <a href="mailto:modesto.redrejo@uam.es" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-envelope-paper"></i></a>
    <a href="https://github.com/StrBio/strbio.github.io" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">StrBio UAM</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Background</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ddbb.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Protein Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pdb.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein Structures</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Modeling</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./homology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Homology Modeling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1d.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1D Features</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./advanced.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Advanced homology modeling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ai.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein modeling in the AlphaFold era</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./links.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Modeling Resources</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul class="collapse">
  <li><a href="#sec-alignment" id="toc-sec-alignment" class="nav-link active" data-scroll-target="#sec-alignment"><span class="header-section-number">1</span> Structure comparison and alignment</a></li>
  <li><a href="#major-protein-databases" id="toc-major-protein-databases" class="nav-link" data-scroll-target="#major-protein-databases"><span class="header-section-number">2</span> Major protein databases</a>
  <ul class="collapse">
  <li><a href="#sequence-databases" id="toc-sequence-databases" class="nav-link" data-scroll-target="#sequence-databases">Sequence databases</a></li>
  <li><a href="#strDDBB" id="toc-strDDBB" class="nav-link" data-scroll-target="#strDDBB">Structure databases</a></li>
  <li><a href="#current-and-future-strategies-in-protein-databases" id="toc-current-and-future-strategies-in-protein-databases" class="nav-link" data-scroll-target="#current-and-future-strategies-in-protein-databases">Current and future strategies in protein databases</a></li>
  </ul></li>
  <li><a href="#practice-1b-protein-database" id="toc-practice-1b-protein-database" class="nav-link" data-scroll-target="#practice-1b-protein-database"><span class="header-section-number">3</span> <span style="color:green"><strong>Practice 1B: Protein Database</strong></span></a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./intro.html">Background</a></li><li class="breadcrumb-item"><a href="./ddbb.html">Protein Databases</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Protein Databases</h1>
</div>



<div class="quarto-title-meta column-body">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 21, 2023</p>
    </div>
  </div>
  
    <div>
    <div class="quarto-title-meta-heading">Modified</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">January 31, 2025</p>
    </div>
  </div>
    
  </div>
  


</header>


<section id="sec-alignment" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Structure comparison and alignment</h1>
<p>To understand the diversity and function of proteins, it is important to compare their sequences and structures. This helps to find common patterns and understand their diversity and evolutionary history. By measuring and analyzing these similarities, scientists can classify proteins and determine their relationships in terms of function and evolution. This process is also crucial in protein modeling, as it helps identify, evaluate, and choose intermediate models</p>
<p>It is essential to clarify the distinction between alignment and superposition, as these terms are frequently conflated in the literature. A *structural alignment* aims to identify similarities and differences between two structures, whereas structure superposition displays the structures based on specific criteria, typically derived from a prior structural alignment. Consequently, superposition seeks to minimize the distance between structures by identifying a transformation that achieves either the lowest root-mean-square deviation (<a href="https://en.wikipedia.org/wiki/Root-mean-square_deviation_of_atomic_positions"><strong>RMSD</strong></a>) or the maximum equivalences within an RMSD cutoff.</p>
<p><span id="eq-RMSD"><span class="math display">\[
RMSD(v,w)=\sqrt{\frac{1}{n} \sum_{i=0}^n{||v_{i}-w_{i}||}^2}
\tag{1}\]</span></span></p>
<p>The RMSD can be calculated for any pair of molecules. In the context of proteins, we often refer to the <em>RMSD of the alpha-carbons</em>. A superior alignment will facilitate a better superposition. Therefore, although alignment and superposition are distinct processes, the RMSD can serve as an indicator of both; the lower the RMSD, the better the alignment/superposition. It is important to note that RMSD is an actual distance measure, not a score. That implies that we only can obtain the RMSD for aligned residues, not for the whole sequence of any of the two proteins. Therefore, an RMSD of 1 Å may indicate a close distance but, if it involves very few amino acids, it does not necessarily suggest good similarity. Both the RMSD value and the number of aligned residues must be considered for accurate analysis.</p>
<div id="fig-rmsd" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rmsd-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="pics/rmsd.png" title="RMSD_example" class="figure img-fluid figure-img" width="500">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rmsd-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Structural alignment of three different AP endoncleases obtained with the <code>cealign</code> algorithm implemented in Pymol. The RMSD and the number of aligned residues for each pair of structures is indicated.
</figcaption>
</figure>
</div>
</section>
<section id="major-protein-databases" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Major protein databases</h1>
<p>Classification of protein sequences helps us understand the diversity of different proteins by examining their sequences, known as the <em>protein sequence space</em> (the mathematical concept of <a href="https://en.wikipedia.org/wiki/Space_(mathematics)">space</a>). On the other hand, classifying protein structures involves grouping proteins based on their structural relationships. Some classifications consider the structural neighborhood (structural continuum), while others use the concept of protein evolution as the main factor of diversification, resulting in a discrete rather than continuous <em>protein structure space</em> <span class="citation" data-cites="andreeva2012">(<a href="#ref-andreeva2012" role="doc-biblioref">Andreeva 2012</a>)</span>.</p>
<p>This section does not intend to provide an exhaustive review of all protein databases. Many of the databases discussed are likely already familiar to the reader. Consequently, we will not cover in detail the <a href="https://www.ncbi.nlm.nih.gov/protein/">NCBI Protein Database</a>, which is widely used for various purposes and is likely mentioned in other courses. The NCBI Protein Database primarily serves as a main repository of sequences, with minimal emphasis on the analysis of protein diversity and classification. This section will highlight the main differences and applications of Pfam, Uniprot, Prosite, PDB, SCOP, and CATH. For more detailed information, please refer to the references cited below.</p>
<div id="fig-ddbb" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ddbb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="pics/ddbb.png" title="Protein databases" class="figure img-fluid figure-img" width="383">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ddbb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Major protein databases. Although most of these databases are around 30 years old (or more), since they were created (or renovated) during the last years of the 20th century or the early 21st.
</figcaption>
</figure>
</div>
<p>In bioinformatics, databases are often classified as primary or secondary. <strong>Primary databases</strong> contain experimentally derived data such as nucleotide sequences, protein sequences, or macromolecular structures. It is important to note that once a database accession number is assigned, the data in primary databases remain unchanged, forming part of the scientific record. In contrast, <strong>secondary databases</strong> include data derived from analyzing primary data. These databases often utilize information from numerous sources, including other databases and the scientific literature. They are typically highly curated and involve a complex combination of computational algorithms and/or manual analysis and interpretation to generate new knowledge from the public record of science.</p>
<p>Although the distinction between primary and secondary databases has become less clear in recent times due to the integration of data from various sources, some differences can still be discerned. The main primary databases for protein sequences include NCBI Protein and RCSB-PDB for protein structures. UniProt also hosts a primary sequence database named TrEMBL and, since 2002, incorporates the PIR-PSD database, which unites the resources of the Protein Information Resource, EMBL, and SIB into a single meta-database (see <a href="https://proteininformationresource.org/pirwww/dbinfo/pir_psd.shtml">PIR-PSD</a>). On the other hand, RCSB-PDB serves as the major structural primary database, while SCOP2 and CATH are notable secondary databases.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Programatic access
</div>
</div>
<div class="callout-body-container callout-body">
<p>All of the databases we describe here allow programmatic and/or API access, commonly with BioPython and R packages, which significantly increases the possibilities for scripting and batch data analysis</p>
</div>
</div>
<div id="fig-full" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-full-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="https://github.com/StrBio/strbio2022_23/raw/main/pics/bbdd_full.pdf"><img src="pics/bbdd_full.png" class="figure img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-full-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Main features of protein databases (updated in 2020)
</figcaption>
</figure>
</div>
<section id="sequence-databases" class="level2">
<h2 class="anchored" data-anchor-id="sequence-databases">Sequence databases</h2>
<section id="uniprot" class="level3">
<h3 class="anchored" data-anchor-id="uniprot">Uniprot</h3>
<p>The <a href="https://www.uniprot.org/">Uniprot</a> databases are managed by the UniProt consortium, established in 2002 by <a href="https://en.wikipedia.org/wiki/European_Bioinformatics_Institute">EMBL-EBI</a>, <a href="https://en.wikipedia.org/wiki/Swiss_Institute_of_Bioinformatics">SIB</a>, and <a href="https://en.wikipedia.org/wiki/Protein_Information_Resource">PIR</a>. Nowadays, UniProt can be considered a metadatabase as its entries contain information from various sources. It was created with two main goals: to establish a comprehensive, non-redundant protein sequence database and to enrich that database with detailed annotations. These annotations include protein and gene families, function and structure-function data, interactions with other proteins or cofactors, localization, expression patterns, variants, and more. Thus, it aims to fulfill the objectives of both primary and secondary databases.</p>
<p>The central hub of the UniProt databases is the <a href="https://www.uniprot.org/help/uniprotkb">Uniprot Knowledgebase</a>. It is a collection of functional information on proteins, with accurate, consistent, and rich annotations. The UniProtKB consists of two internal databases: one section contains manually-annotated records with information drawn from literature, community suggestions, and curator-reviewed computational analysis. The other section includes computationally analyzed records. These sections are referred to as “UniProtKB/Swiss-Prot” (reviewed, manually annotated) and “UniProtKB/TrEMBL” (unreviewed, automatically annotated), respectively.</p>
<p>In recent years, UniProtKB has incorporated structural data from the <a href="https://alphafold.ebi.ac.uk/">Alphafold</a> database, in addition to cross-references to structural information. See the module <a href="ai.html">“State-of-the-art protein modeling with Deep Learning-based methods”</a> for more information.</p>
<p>UniProt contains sequences with varying levels of annotation detail in two complementary databases: <a href="https://www.uniprot.org/help/uniparc">Uniparc</a> and <a href="https://www.uniprot.org/help/uniref">Uniref</a>. Briefly, UniParc (UniProt Archive) is a comprehensive and non-redundant database that includes most publicly available protein sequences worldwide. UniParc avoids redundancy by storing each unique sequence only once and assigning it a stable, unique identifier (UPI), which allows the identification of the same protein from different source databases. A UPI is never removed, changed, or reassigned. On the other hand, UniRef (UniProt Reference Clusters) provides clustered sets of sequences from UniProtKB (and selected UniParc records) to ensure complete coverage of the sequence space at various resolutions while hiding redundant sequences (but not their descriptions) from view. The UniRef100 database combines identical sequences into a single UniRef entry, displaying the sequence of a representative protein, the accession numbers of all merged entries, and links to the corresponding databases. UniRef90 is built by clustering UniRef100 sequences using the MMseqs2 algorithm <span class="citation" data-cites="steinegger2018">(<a href="#ref-steinegger2018" role="doc-biblioref">Steinegger and Söding 2018</a>)</span>, so each cluster consists of sequences with at least 90% sequence identity and 80% overlap with the longest sequence (the <a href="https://www.uniprot.org/help/uniref_seed"><strong>seed sequence</strong></a> ) of the cluster. Similarly, UniRef50 is built by clustering UniRef90 seed sequences that have at least 50% sequence identity and 80% overlap with the longest sequence in the cluster. UniParc and UniRef contain only protein sequences; all other information about the proteins must be retrieved from the source databases using database cross-references.</p>
</section>
<section id="pfam" class="level3">
<h3 class="anchored" data-anchor-id="pfam">Pfam</h3>
<p>Pfam is a protein database that aims to classify sequences by their evolutionary relationships. It was founded in 1995 and it has been very useful for functional annotation of genomic data. Pfam’s website (<a href="http://pfam.xfam.org/" class="uri">http://pfam.xfam.org/</a>) was closed by the end of 2022. However, Pfam database was not discontinued, but integrated into the <strong>InterPro</strong> site (see the <a href="https://xfam.wordpress.com/2022/08/04/pfam-website-decommission/">Xfam Blog entry</a>).</p>
<p>Pfam uses <a href="https://www.ebi.ac.uk/training/online/courses/protein-classification-intro-ebi-resources/what-are-protein-signatures/signature-types/what-are-hmms/">HMM</a> profiles to classify proteins into families, which are grouped into clans. Check out the EBI course about using Pfam: <a href="https://www.ebi.ac.uk/training/online/courses/pfam-creating-protein-families/" class="uri">https://www.ebi.ac.uk/training/online/courses/pfam-creating-protein-families/</a></p>
<p>The current release (37.1) contains 23.794 entries and 751 clans. Pfam was designed as a database that should be often updated in the fast-forward genomic era. To this aim, they use two alignment types. Each Pfam family has a seed alignment that contains a representative set of sequences for the entry. A profile hidden Markov model (HMM) is automatically built from the seed alignment and searched against a sequence database called <em>pfamseq</em> using the HMMER3 software (<a href="http://hmmer.org/" class="uri">http://hmmer.org/</a>). All sequence regions that satisfy a family-specific curated threshold, also known as the gathering threshold, are aligned to the profile HMM to create the full alignment <span class="citation" data-cites="mistry2020">(<a href="#ref-mistry2020" role="doc-biblioref">Mistry et al. 2020</a>)</span>.</p>
<p>In addition to the HMM-based Pfam entries (<strong>Pfam-A</strong>), the Pfam profiles are used to provide a set of unannotated, computationally generated multiple sequence alignments called <strong>Pfam-B</strong>. However, in the last Pfam versions, the Pfam-B alignments are presently only released on the Pfam FTP site.</p>
<p>Pfam has also been used in the creation of other resources such as <a href="https://rfam.org/">Rfam</a> (RNA families) and <a href="https://www.dfam.org/">Dfam</a> (transposable DNA elements).</p>
</section>
<section id="interpro" class="level3">
<h3 class="anchored" data-anchor-id="interpro">InterPro</h3>
<p><a href="https://www.ebi.ac.uk/interpro/">InterPro</a> aims to be a functional secondary database, by classifying proteins into families, domains, and important sites. To classify proteins in this way, InterPro uses predictive models, known as signatures, provided by several different databases (up to 13) that make up the InterPro consortium. InterPro combines those different signatures representing equivalent families, domains, or sites, and provides additional information such as descriptions, literature references, and Gene Ontology (GO) terms, to produce a comprehensive resource for protein classification (see <span class="citation" data-cites="blum2021">Blum et al. (<a href="#ref-blum2021" role="doc-biblioref">2021</a>)</span>).</p>
<p>InterPro database is updated every 2 months and it is very useful for <strong>annotation</strong> of ORFans or divergent proteins. In the last years, it has integrated more resources, including Pfam, as well as structural data and predictions, giving rise to a very handy resource for multiple purposes in protein science <span class="citation" data-cites="paysan-lafosse2023">(<a href="#ref-paysan-lafosse2023" role="doc-biblioref">Paysan-Lafosse et al. 2023</a>)</span>.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>Interpro was created as a sequence DDBB, but it is currently in between. I’d rather say that it is more a “meta-database” containing sequence and structure information.</strong></p>
</div>
</div>
</section>
</section>
<section id="strDDBB" class="level2">
<h2 class="anchored" data-anchor-id="strDDBB">Structure databases</h2>
<section id="rcsb-pdb" class="level3">
<h3 class="anchored" data-anchor-id="rcsb-pdb">RCSB-PDB</h3>
<p>The Protein Data Bank (PDB, <a href="www.rcsb.org">www.rcsb.org</a>) database is the major macromolecule structural primary database <span class="citation" data-cites="burley2021">(<a href="#ref-burley2021" role="doc-biblioref">Burley et al. 2020</a>)</span>. It contains mostly protein structures, but also spans nucleic acids and nucleoprotein complexes. PDB turned 50 in 2021 and you can see a detailed overview of its <a href="https://www.rcsb.org/pages/about-us/history">history in the RCSB-PDB site.</a></p>
<p>Briefly, the PDB was established in 1971 at Brookhaven National Laboratory with only 7 structures. Then, the <strong>Research Collaboratory for Structural Bioinformatics</strong> (RCSB), formed by Rutgers, UCSD/SDSC, and CARB/NIST, became responsible for the management of the PDB in 1998 in response to an RFP and a lengthy review process.&nbsp;In 2003, the Worldwide Protein Data Bank (<a href="http://wwpdb.org/">wwPDB</a>) was formed to maintain a single PDB archive of macromolecular structural data that is freely and publicly available to the global community. It consists of organizations that act as deposition, data processing and distribution centers for PDB data.</p>
<p>PDB structures are largely obtained by X-ray crystallography, but it accepts derived from EM and RMN data since 1989 and 1991, respectively. Indeed, the BMRB (<a href="https://bmrb.io/">Biological Magnetic Resonance Bank</a>) has partnered with the PDB since 2006 and the EMBD (<a href="https://www.ebi.ac.uk/emdb/">Electron Microscopy Data Bank</a>) since 2021. Furthermore, <a href="https://www.rcsb.org/news/6304ee57707ecd4f63b3d3db">starting September 2022</a>, the PDB also contains computed models from the <a href="https://alphafold.ebi.ac.uk/">Alphafold database</a> (which we will discuss later in this course) and <a href="https://www.modelarchive.org/">RoseTTAFold-ModelArchive</a>. <strong>Thus, the PDB database is the major hub that centralizes biological structures nowadays.</strong></p>
<p>PDB database has four mirrors and websites (<a href="www.rcsb.org">RCSB</a>, <a href="https://www.ebi.ac.uk/pdbe/">Europe</a>, <a href="https://bmrb.io/">BMRB</a>, and <a href="https://pdbj.org/">Japan</a>) with mainly overlapping information, although they have some specialization. The RCSB PDB site has also an educational section (<a href="https://pdb101.rcsb.org/">PDB-101</a>) with very useful info and resources for teaching and learning structural biology and the work with PDB structures.</p>
<p>The PDB entries contain all the information about the structure, spanning from the protein sequence and source to the experiment details, as well as the structure assessment (see <a href="pdb.html#sec-assess">Structural quality assurance</a> section) and visualization. You can download all this information and the structure coordinates in diverse file formats.</p>
</section>
<section id="scop" class="level3">
<h3 class="anchored" data-anchor-id="scop">SCOP</h3>
<p>The Structural Classification of Proteins (SCOP, <a href="http://scop.mrc-lmb.cam.ac.uk/">http://scop.mrc-lmb.cam.ac.uk</a>) database is a <em>classification of protein domains</em> organized according to their evolutionary and structural relationships in hierarchical categories. The main unit is the <em>family</em> that groups related proteins with clear evidence for their evolutionary origin while the <em>superfamily</em> brings together more distantly related protein domains. Further, superfamilies are grouped into distinct <em>folds</em> on the basis of the global structural features shared by the majority of their members. Domain definitions are provided for the two main levels of the SCOP classification, family and superfamily, and the domain boundaries for each of these can coincide or differ <span class="citation" data-cites="andreeva2020">(<a href="#ref-andreeva2020" role="doc-biblioref">Andreeva et al. 2020</a>)</span>.</p>
<p>For each group, a representative is selected based on its sequence (UniProtKB) and structure (PDB) and used for SCOP classification. Thus, the SCOP domain boundaries are assigned to both, the PDB and UniProtKB entry.</p>
</section>
<section id="cath" class="level3">
<h3 class="anchored" data-anchor-id="cath">CATH</h3>
<p>CATH (<a href="www.cathdb.info">www.cathdb.info</a>) is a free, publicly available resource that identifies protein domains within proteins from the Protein Data Bank and classifies them into evolutionary-related groups according to sequence, structure, and function information. It assumes that related proteins which fold alike often exhibit similar functions (this only could be demonstrated if we find intermediates). CATH uses a hierarchical classification scheme where the units compared and classified are structural domains. Domains, defined here as globular structural domains capable of semi-independent folding, are extracted from experimentally determined protein structures available in the PDB database. The domains are classified into the following hierarchical levels that compose the CATH name: Class (<strong>C</strong>), Architecture (<strong>A</strong>), Topology (<strong>T</strong>), and Homologous superfamilies (<strong>H</strong>).</p>
<p>CATH uses a combination of several structure-based (SSAP, CATHEDRAL) and sequence-based (Needleman-Wunsch-based sequence alignments, Jackhmmer, Profile Comparer, and HHsearch) algorithms to assess the similarity of domains to each other and to identify protein homologues <span class="citation" data-cites="sillitoe2021">(<a href="#ref-sillitoe2021" role="doc-biblioref">Sillitoe et al. 2021</a>)</span>.</p>
<p>CATH has a sister resource, Gene3D, that adds in additional protein domain sequences with no known structure, which brings the current total number of domains in CATH-Gene3D up to 95 million.</p>
<p>CATH database is updated quite regularly through daily snapshots (CATH-B), but a full release with more tools, named CATH+ is released every 12 months. CATH-plus contains functional families (CATH-FunFams), structural clusters, and other tools <span class="citation" data-cites="sillitoe2021">(<a href="#ref-sillitoe2021" role="doc-biblioref">Sillitoe et al. 2021</a>)</span>.</p>
</section>
</section>
<section id="current-and-future-strategies-in-protein-databases" class="level2">
<h2 class="anchored" data-anchor-id="current-and-future-strategies-in-protein-databases">Current and future strategies in protein databases</h2>
<p>There is a significant trend towards the cross-referencing and integration of diverse data within <em>metadatabases</em>. An exemplary case is the <a href="https://www.proteinatlas.org/">Human Protein Atlas</a> which provides information about proteins classified by cell type or tissue, along with details on splicing variants, mutants, and more. Additionally, it is important to acknowledge the new structural databases, such as Deepmind’s <a href="https://alphafold.ebi.ac.uk/">Alphafold Database</a> and the <a href="https://esmatlas.com/">ESM Metagenomic Atlas</a>, which house millions of protein structures predicted using deep-learning methods. There are also specialized databases, like BFDV, that contain viral protein structures obtained through Alphafold and can be searched using <a href="https://search.foldseek.com/">Foldseek</a>, a method designed to identify structural similarities.</p>
<p>Given the recent boost in the capacity for accurately obtain protein models, some author suggested (or desired) that future databases contain not only protein sequence variants and protein complexes, but also diverse conformations for each structure, which would help to gain insights into their function and biological role <span class="citation" data-cites="beck2024">(<a href="#ref-beck2024" role="doc-biblioref">Beck et al. 2024</a>)</span>.</p>
</section>
</section>
<section id="practice-1b-protein-database" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> <span style="color:green"><strong>Practice 1B: Protein Database</strong></span></h1>
<p>The goal of this exercise is to get familiar with protein structures and their domain organization. First just try to explore the structures and identify similarities and differences by naked eye.</p>
<div class="columns">
<div class="column" style="width:15%;">
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: center;">Structures</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">2MNA</td>
</tr>
<tr class="even">
<td style="text-align: center;">2WB0</td>
</tr>
<tr class="odd">
<td style="text-align: center;">1C0A</td>
</tr>
<tr class="even">
<td style="text-align: center;">1QUM</td>
</tr>
<tr class="odd">
<td style="text-align: center;">5ZHZ</td>
</tr>
<tr class="even">
<td style="text-align: center;">5CFE</td>
</tr>
<tr class="odd">
<td style="text-align: center;">1DE9</td>
</tr>
<tr class="even">
<td style="text-align: center;">6WCD</td>
</tr>
<tr class="odd">
<td style="text-align: center;">1AKO</td>
</tr>
<tr class="even">
<td style="text-align: center;">6KHY</td>
</tr>
<tr class="odd">
<td style="text-align: center;">1EBM</td>
</tr>
</tbody>
</table>
</div>
</div>
<section id="use-structural-databases-cath-scop2-or-pdb-to-classify-the-proteins." class="level3">
<h3 class="anchored" data-anchor-id="use-structural-databases-cath-scop2-or-pdb-to-classify-the-proteins.">1. Use structural databases (CATH, Scop2 or PDB) to classify the proteins.</h3>
<p>First, describe the structure in your own words. Then, using generic terms applicable to any hierarchical classification system, classify this small dataset of structures.</p>
</section>
<section id="answer-the-following-questions" class="level3">
<h3 class="anchored" data-anchor-id="answer-the-following-questions">2. Answer the following questions</h3>
<p>No need of long answers, ~100 words each question would be fine.</p>
<section id="a.-can-you-suggest-the-structural-and-evolutionary-relationship-between-them" class="level4">
<h4 class="anchored" data-anchor-id="a.-can-you-suggest-the-structural-and-evolutionary-relationship-between-them"><strong>a. Can you suggest the structural and evolutionary relationship between them?</strong></h4>
</section>
<section id="b.-how-many-domains-are-in-each-protein-in-scop2-and-cath-are-the-numbers-the-same-whywhy-not" class="level4">
<h4 class="anchored" data-anchor-id="b.-how-many-domains-are-in-each-protein-in-scop2-and-cath-are-the-numbers-the-same-whywhy-not"><strong>b. How many domains are in each protein in Scop2 and Cath? Are the numbers the same? Why/why not?</strong></h4>
</section>
<section id="c.-which-are-the-proscons-of-each-database" class="level4">
<h4 class="anchored" data-anchor-id="c.-which-are-the-proscons-of-each-database"><strong>c.&nbsp;Which are the pros/cons of each database?</strong></h4>
</section>
<section id="references" class="level4">




</section>
</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-andreeva2012" class="csl-entry" role="listitem">
Andreeva, Antonina. 2012. <span>“Classification of Proteins: Available Structural Space for Molecular Modeling.”</span> In, edited by Andrew J. W. Orry and Ruben Abagyan, 1–31. Totowa, NJ: Humana Press. <a href="https://doi.org/10.1007/978-1-61779-588-6_1">https://doi.org/10.1007/978-1-61779-588-6_1</a>.
</div>
<div id="ref-andreeva2020" class="csl-entry" role="listitem">
Andreeva, Antonina, Eugene Kulesha, Julian Gough, and Alexey G. Murzin. 2020. <span>“The SCOP database in 2020: expanded classification of representative family and superfamily domains of known protein structures.”</span> <em>Nucleic Acids Research</em> 48 (D1): D376–82. <a href="https://doi.org/10.1093/nar/gkz1064">https://doi.org/10.1093/nar/gkz1064</a>.
</div>
<div id="ref-beck2024" class="csl-entry" role="listitem">
Beck, Martin, Roberto Covino, Inga Hänelt, and Michaela Müller-McNicoll. 2024. <span>“Understanding the cell: Future views of structural biology.”</span> <em>Cell</em> 187 (3): 545–62. <a href="https://doi.org/10.1016/j.cell.2023.12.017">https://doi.org/10.1016/j.cell.2023.12.017</a>.
</div>
<div id="ref-blum2021" class="csl-entry" role="listitem">
Blum, Matthias, Hsin-Yu Chang, Sara Chuguransky, Tiago Grego, Swaathi Kandasaamy, Alex Mitchell, Gift Nuka, et al. 2021. <span>“The InterPro Protein Families and Domains Database: 20 Years On.”</span> <em>Nucleic Acids Research</em> 49 (D1): D344–54. <a href="https://doi.org/10.1093/nar/gkaa977">https://doi.org/10.1093/nar/gkaa977</a>.
</div>
<div id="ref-burley2021" class="csl-entry" role="listitem">
Burley, Stephen K, Charmi Bhikadiya, Chunxiao Bi, Sebastian Bittrich, Li Chen, Gregg V Crichlow, Cole H Christie, et al. 2020. <span>“<span class="nocase">RCSB Protein Data Bank: powerful new tools for exploring 3D structures of biological macromolecules for basic and applied research and education in fundamental biology, biomedicine, biotechnology, bioengineering and energy sciences</span>.”</span> <em>Nucleic Acids Research</em> 49 (D1): D437–51. <a href="https://doi.org/10.1093/nar/gkaa1038">https://doi.org/10.1093/nar/gkaa1038</a>.
</div>
<div id="ref-mistry2020" class="csl-entry" role="listitem">
Mistry, Jaina, Sara Chuguransky, Lowri Williams, Matloob Qureshi, Gustavo&nbsp;A Salazar, Erik L L Sonnhammer, Silvio C E Tosatto, et al. 2020. <span>“Pfam: The Protein Families Database in 2021.”</span> <em>Nucleic Acids Research</em> 49 (D1): D412–19. <a href="https://doi.org/10.1093/nar/gkaa913">https://doi.org/10.1093/nar/gkaa913</a>.
</div>
<div id="ref-paysan-lafosse2023" class="csl-entry" role="listitem">
Paysan-Lafosse, Typhaine, Matthias Blum, Sara Chuguransky, Tiago Grego, Beatriz Lázaro Pinto, Gustavo&nbsp;A Salazar, Maxwell&nbsp;L Bileschi, et al. 2023. <span>“InterPro in 2022.”</span> <em>Nucleic Acids Research</em> 51 (D1): D418–27. <a href="https://doi.org/10.1093/nar/gkac993">https://doi.org/10.1093/nar/gkac993</a>.
</div>
<div id="ref-sillitoe2021" class="csl-entry" role="listitem">
Sillitoe, Ian, Nicola Bordin, Natalie Dawson, Vaishali P. Waman, Paul Ashford, Harry M. Scholes, Camilla S. M. Pang, et al. 2021. <span>“CATH: increased structural coverage of functional space.”</span> <em>Nucleic Acids Research</em> 49 (D1): D266–73. <a href="https://doi.org/10.1093/nar/gkaa1079">https://doi.org/10.1093/nar/gkaa1079</a>.
</div>
<div id="ref-steinegger2018" class="csl-entry" role="listitem">
Steinegger, Martin, and Johannes Söding. 2018. <span>“Clustering huge protein sequence sets in linear time.”</span> <em>Nature Communications</em> 9 (1): 2542. <a href="https://doi.org/10.1038/s41467-018-04964-5">https://doi.org/10.1038/s41467-018-04964-5</a>.
</div>
</div></section></div></main> <!-- /main -->
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
          // target, if specified
          link.setAttribute("target", "_blank");
          if (link.getAttribute("rel") === null) {
            link.setAttribute("rel", "noopener");
          }
          // default icon
          link.classList.add("external");
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation column-body">
  <div class="nav-page nav-page-previous">
      <a href="./intro.html" class="pagination-link" aria-label="Introduction">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Introduction</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./pdb.html" class="pagination-link" aria-label="Protein Structures">
        <span class="nav-page-text">Protein Structures</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>Modesto Redrejo Rodríguez, 2025</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://creativecommons.org/licenses/by-nc/2.0/">
      <i class="bi bi-badge-cc-fill" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://twitter.com/mredrejo">
<p><iconify-icon inline="" icon="fa6-brands:x-twitter"></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://bsky.app/profile/mredrejo.bsky.social">
<p><iconify-icon inline="" icon="simple-icons:bluesky"></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link active" href="https://github.com/StrBio/strbio.github.io" aria-current="page">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>




<script src="site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>