<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-08-21">

<title>Structural Bioinformatics at UAM - Introduction</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./ddbb.html" rel="next">
<link href="./index.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/iconify-1.0.0-beta.2/iconify-icon.min.js"></script>
<script src="site_libs/quarto-contrib/molstar-v3.13.0/molstar.js"></script>
<link href="site_libs/quarto-contrib/molstar-v3.13.0/molstar.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/molstar-v3.13.0/app-container.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>


<link rel="stylesheet" href="styles.css">
<link rel="stylesheet" href="include/webex.css">
<meta name="twitter:title" content="Structural Bioinformatics at UAM - Introduction">
<meta name="twitter:description" content="">
<meta name="twitter:image" content="pics/Protein_structure.png">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./intro.html">Background</a></li><li class="breadcrumb-item"><a href="./intro.html">Introduction</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Structural Bioinformatics at UAM</a> 
        <div class="sidebar-tools-main">
    <a href="mailto:modesto.redrejo@uam.es" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-envelope-paper"></i></a>
    <a href="https://github.com/StrBio/strbio.github.io" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">StrBio 2023-24</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Background</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ddbb.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pdb.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein Structures</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Modeling</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./homology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Homology Modeling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1d.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1D Features</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./advanced.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Advanced homology modeling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ai.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein modeling in the AlphaFold era</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./links.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Modeling Resources</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul class="collapse">
  <li><a href="#goals" id="toc-goals" class="nav-link active" data-scroll-target="#goals"><span class="header-section-number">1</span> Goals</a></li>
  <li><a href="#sec-str" id="toc-sec-str" class="nav-link" data-scroll-target="#sec-str"><span class="header-section-number">2</span> Before going forward: Protein Structure 101</a>
  <ul class="collapse">
  <li><a href="#sec-rama" id="toc-sec-rama" class="nav-link" data-scroll-target="#sec-rama"><span class="header-section-number">2.1</span> Ramachandran Plot</a></li>
  <li><a href="#protein-folds-domains-and-motifs" id="toc-protein-folds-domains-and-motifs" class="nav-link" data-scroll-target="#protein-folds-domains-and-motifs"><span class="header-section-number">2.2</span> Protein folds, domains and motifs</a></li>
  <li><a href="#quick-exercise" id="toc-quick-exercise" class="nav-link" data-scroll-target="#quick-exercise"><span class="header-section-number">2.3</span> <span style="color:green"><strong>Quick exercise</strong></span></a></li>
  <li><a href="#hands-on-playing-with-secondary-structures" id="toc-hands-on-playing-with-secondary-structures" class="nav-link" data-scroll-target="#hands-on-playing-with-secondary-structures"><span class="header-section-number">2.4</span> <span style="color:green"><strong>Hands on: Playing with secondary structures</strong></span></a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Introduction</h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 21, 2023</p>
    </div>
  </div>
  
    <div>
    <div class="quarto-title-meta-heading">Modified</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">August 30, 2023</p>
    </div>
  </div>
    
  </div>
  

</header>

<section id="goals" class="level1" data-link="Preface" data-number="1">
<h1 data-link="Preface" data-number="1"><span class="header-section-number">1</span> Goals</h1>
<p><a href="https://en.wikipedia.org/wiki/Structural_bioinformatics" title="Wikipedia">Structural Bioinformatics</a> (SB) is a broad discipline that comprises data resources, algorithms, and tools for investigating, analyzing, predicting, and interpreting biomacromolecular structures <span class="citation" data-cites="paiva2022">Paiva et al. (<a href="#ref-paiva2022" role="doc-biblioref">2022</a>)</span>. More specifically, we are interested in protein structural bioinformatics, spanning visualization and analysis of the structure of biomacromolecules as well as the prediction of protein structures and complexes. The great promise of SB is predicated on the belief that a high-resolution structural information about biological systems will allow us to precisely reason about the function of these systems and the effects of modifications and perturbations.</p>
<p>The goals of SB require at least four different research lines (see Chapter 1 in <span class="citation" data-cites="structur">Gu and Bourne (<a href="#ref-structur" role="doc-biblioref">2009</a>)</span>):</p>
<ol type="1">
<li><p><em>Visualization</em> of complex structures with several sources of information: sequence, structural data, electrostatic fields, location of functional sites, and areas of variability.</p></li>
<li><p><em>Classification</em> of the structures, making if necessary to cluster similar structures together in a hierarchical classification allow us to identify common origins and diversification paths. Similar to other fields of biology classification is tedious but required to understand the structural space.</p></li>
<li><p><em>Prediction</em> of structures remains an area of keen interest and a field of research itself. As we will see below, the number of different sequences is much higher than the availability of structures, which make prediction an essential and useful tool.</p></li>
<li><p><em>Simulation.</em> Experimentally obtained structures are primarily static structural models (see warning below). However, the properties of these molecules are often the results of their dynamic motions. The definition of energy functions that govern the folding of proteins and their subsequent stable dynamics can be analyzed by molecular dynamics simulations, although computation capacities may be limiting to reach a biologically relevant timescales.</p></li>
</ol>
<p>Powered by large amount of data and great technical advances, the field has experienced a great revolution in the last decade. The increase of experimental capacities to analyze the structure of proteins and other biological molecules and structures (see <span class="citation" data-cites="callaway2020">Callaway (<a href="#ref-callaway2020" role="doc-biblioref">2020</a>)</span>) and the development of Artificial Intelligence (AI)-assisted structure prediction boosted the capacity of life-science researchers to address a wide variety of questions regarding proteins diversity, evolution and function. This revolution underwent a great acceleration in the last 2-3 years and the implications in biology, biotechnology, and biomedicine are still unforeseen.</p>
</section>
<section id="sec-str" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Before going forward: Protein Structure 101</h1>
<p>Although you can make some protein modeling without being an expert in structural biology, a basic understanding of protein structure is strongly advisable. In this course there are some students without a background in biology. Moreover, over the years, I noticed that graduate students in biology, biomedicine, and related fields have a very different background on protein structure. If you want to review and update your background on protein structure, I recommend you reading Chapter 2 of <span class="citation" data-cites="structur">Gu and Bourne (<a href="#ref-structur" role="doc-biblioref">2009</a>)</span>, the great recent review by <span class="citation" data-cites="stollar2020">Stollar and Smith (<a href="#ref-stollar2020" role="doc-biblioref">2020</a>)</span> and the <a href="https://en.wikipedia.org/wiki/Protein_structure">Wikipedia</a> and <a href="https://proteopedia.org/wiki/index.php/Introduction_to_protein_structure">Proteopedia</a> articles on protein structures, which constituted my main source for this brief section (follow picture links).</p>
<div id="fig-str" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://en.wikipedia.org/wiki/Protein_structure"><img src="pics/Protein_structure.png" title="Protein Structure" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">Figure&nbsp;1: Protein structure levels, using human PCNA (PDB 1AXC) as an example.</figcaption>
</figure>
</div>
<p>Proteins are key components of life, playing key roles in almost any possible vital function, either as structural, or scaffolding elements or as active enzymes that catalyze metabolic reactions. Proteins are built as polymers of amino acids and the sequence of amino acids of a particular protein can be also called the <strong>primary structure</strong> of the protein. Amino acid chains can spontaneously fold up into three-dimensional structures, mostly stabilized by hydrogen bonds between amino acids. The amino acid sequence determines different layers of 3D structure. Each of the 20 natural amino acids has different physicochemical properties that affect its preferred conformation. Thus, the first level of folding is called <strong>secondary structure</strong>, forming common patterns as we will see in a moment.</p>
<div id="fig-aa" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://www.reddit.com/r/chemistry/comments/acyald/venn_diagram_showing_the_properties_of_the_20/"><img src="pics/aa.png" title="Amino acids clasification by type" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">Figure&nbsp;2: Amino acids classification by type</figcaption>
</figure>
</div>
<p>These stretches of secondary structure patterns can fold in 3D due to interactions between the side chains of amino acids. This is called protein <strong>Tertiary structure</strong>. Finally, two or more individual peptide chains can form multisubunit proteins that have the so-called <strong>Quaternary structure</strong>.</p>
<p>It should be noted that the peptide bond itself cannot rotate as it has a double bond-like character. Therefore, rotation can only occur about the bond between the Cα and the C = O group, (the phi (φ) angle) and the Cα and the NH group, (the psi (ψ) angle). In fact, the polypeptide backbone chain is composed of a repeating series of two rotatable bonds followed by one non-rotatable (peptide) bond. However, not all 360º of the psi and phi angles are possible as neighboring sidechains can clash due to steric hindrance. For certain angles and amino acid combinations, the atoms cannot be in the same physical place and this partly explains why some amino acids have a higher propensity (likelihood) to form different types of secondary structures.</p>
<div id="fig-bond" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://portlandpress.com/essaysbiochem/article/64/4/649/226515/Uncovering-protein-structure"><img src="pics/peptide_bond.png" title="Peptide bond" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">Figure&nbsp;3: Scheme of a generic polypeptide chain. Residue side chains are denoted as R. Coloured rectangles indicate sets of six atoms that are coplanar due to the double-bond character of the peptide bond. Arrows indicate the bonds that are free to rotate with the angle of rotation about the N–Cα known as phi and about the Cα–C known as psi. Note that only peptide backbone bonds are labeled and in most cases the R group bond is free to rotate.</figcaption>
</figure>
</div>
<div id="ss">
<p>Within these restraints, the two principal local conformations that avoid steric hindrance and maximize backbone–backbone hydrogen bonding are the <strong>α-helix</strong> and the <strong>β-sheet</strong> secondary structures. The α-helix was proposed initially as left-handed by <a href="https://en.wikipedia.org/wiki/Linus_Pauling">Linus Pauling</a> in 1951, but the crystal structure of myoglobin in 1958 showed that, although both can be found, the right-handed form is the common one. In the common right-handed helices, the backbone NH group hydrogen bonds to the backbone C = O group of the amino acid located four residues earlier along the protein sequence. This results in a polypeptide chain that twists in a regular coil shape with the R-groups pointing outwards away from the peptide backbone. It takes approximately 3.6 residues to complete a full turn of a helix.</p>
</div>
<div class="quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div id="fig-alpha" class="quarto-figure quarto-figure-center anchored" style="flex-basis: 100.0%;justify-content: center;">
<figure class="figure">
<p><a href="https://en.wikipedia.org/wiki/Alpha_helix"><img src="pics/alpha.png" title="Alpha helix" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">Figure&nbsp;4: Alpha helix.</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div id="fig-beta" class="quarto-figure quarto-figure-center anchored" style="flex-basis: 100.0%;justify-content: center;">
<figure class="figure">
<p><a href="https://en.wikipedia.org/wiki/Beta_sheet"><img src="pics/beta.png" title="Beta sheet" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">Figure&nbsp;5: Detailed description of a beta sheet made up of three beta strands.</figcaption>
</figure>
</div>
</div>
</div>
<p>Different amino-acid sequences have different propensities for forming α-helical structures. <a href="https://en.wikipedia.org/wiki/Methionine" title="Methionine">Methionine</a>, <a href="https://en.wikipedia.org/wiki/Alanine" title="Alanine">alanine</a>, <a href="https://en.wikipedia.org/wiki/Leucine" title="Leucine">leucine</a>, <a href="https://en.wikipedia.org/wiki/Glutamate" title="Glutamate">glutamate</a>, and <a href="https://en.wikipedia.org/wiki/Lysine" title="Lysine">lysine</a> have especially high helix-forming propensities, whereas <a href="https://en.wikipedia.org/wiki/Proline" title="Proline">proline</a> and <a href="https://en.wikipedia.org/wiki/Glycine" title="Glycine">glycine</a> have poor helix-forming propensities. <a href="https://en.wikipedia.org/wiki/Proline" title="Proline">Proline</a> either breaks or kinks a helix, both because it cannot donate an amide <a href="https://en.wikipedia.org/wiki/Hydrogen_bond" title="Hydrogen bond">hydrogen bond</a> (having no amide hydrogen), and also because its bulky sidechain interferes sterically with the backbone of the preceding turn. <a href="https://en.wikipedia.org/wiki/Glycine" title="Glycine">Glycine</a> is also an alpha helix breaker. Glycine has only a hydrogen as its R-group and therefore has no constrains for its Φ and Ψ angles. This makes it entropically expensive to adopt the relatively constrained α-helical structure. In other words, glycine is too flexible to hold the structure needed for an alpha helix.</p>
<p><strong>β-Sheets</strong> are composed of two or more extended polypeptide chains called β-strands that run alongside each other. They can be arranged in either a parallel or antiparallel manner. The residues arrange themselves in a regular zigzag manner with the adjacent peptide bonds pointing in opposite directions. In this arrangement, the NH group and the C = O group of each amino acid are hydrogen-bonded to the C = O group and NH group respectively on the adjacent strands. Chains can run in opposite directions, forming an antiparallel β-sheet, or in the same direction, forming a parallel β-sheet. Sidechains from each of the residues point away from the sheets and alternate in opposite directions between residues. It is common to see a pattern of alternating hydrophilic and hydrophobic residues in the primary structure, giving the β-sheets hydrophilic and hydrophobic faces.</p>
<p>Large aromatic residues (<a href="https://en.wikipedia.org/wiki/Tyrosine" title="Tyrosine">tyrosine</a>, <a href="https://en.wikipedia.org/wiki/Phenylalanine" title="Phenylalanine">phenylalanine</a>, <a href="https://en.wikipedia.org/wiki/Tryptophan" title="Tryptophan">tryptophan</a>) and β-branched amino acids (<a href="https://en.wikipedia.org/wiki/Threonine" title="Threonine">threonine</a>, <a href="https://en.wikipedia.org/wiki/Valine" title="Valine">valine</a>, <a href="https://en.wikipedia.org/wiki/Isoleucine" title="Isoleucine">isoleucine</a>) are favored to be found in β-strands. As in the case of α-helixes, β-strands are often ended by <a href="https://en.wikipedia.org/wiki/Glycine" title="Glycine">glycines</a>, which are especially common in β-turns (the most common connector between strands), as <a href="https://en.wikipedia.org/wiki/Amino_acid" title="Amino acid">amino acids</a> with positive φ angles.</p>
<p>The side chain of amino acids also have their torsion angles, referred as χ1, χ2, χ3…</p>
<div id="fig-chi" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-306"><img src="images/paste-6EE800AE.png" class="img-fluid figure-img" width="450"></a></p>
<figcaption class="figure-caption">Figure&nbsp;6: Dihedral angles in glutamate: Dihedral angles are the main degrees of freedom for the backbone (ϕ and ψ angles) and the side chain (χ angles) of an amino acid. The number of χ angles varies between zero and four for the 20 standard amino acids. The figure shows a ball-and-stick representation of glutamate, which has three χ angles (from <span class="citation" data-cites="harder2010">Harder et al. (<a href="#ref-harder2010" role="doc-biblioref">2010</a>)</span>).</figcaption>
</figure>
</div>
<section id="sec-rama" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="sec-rama"><span class="header-section-number">2.1</span> Ramachandran Plot</h2>
<p>As you probably already figure out, many combinations of φ and ψ angles are forbidden because of the principle of steric exclusion: two atoms cannot be in the same place at the same time. This was initially shown by <a href="https://en.wikipedia.org/wiki/G._N._Ramachandran">Gopalasamudram Ramachandran</a>, who also devised a plot to visualize the allowed angle values, so-called Ramachandran plot. This plot can represent the angles of a particular amino acid, of all the amino acids in a protein or many proteins. Analysis of φ and ψ angles in known proteins clearly show that roughly three-quarters of all possible φ, ψ combinations are excluded.</p>
<div id="fig-rama0" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://proteopedia.org/wiki/index.php/Ramachandran_Plot"><img src="pics/Ramachandran_plot_general_100K.jpg" title="General Ramachandran plot" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">Figure&nbsp;7: General Ramachandran plot. The density of points reflects how likely is each angle combination, defining the core (red line) and tolerance (orange) regions.</figcaption>
</figure>
</div>
<p>The core regions in the Rama plot also correspond with common secondary structures, as usually represented in textbooks.</p>
<div id="fig-ram" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pics/rama2.png" title="Textbook Rama plot" class="figure img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;8: Definition of secondary structure alternatives by their combination of phi, psi angles.</figcaption>
</figure>
</div>
<p>Functionally and structurally relevant residues are more likely than others to have torsion angles that can be distributed into the <u>allowed but disfavored</u> regions of a Ramachandran plot. The specific geometry of these functionally relevant residues, while somewhat energetically unfavorable, may be important for the protein’s function, catalytic or otherwise. Such conformations need to be stabilized by the protein using H-bonds, steric packing, or other means, and should very seldom occur for highly solvent-exposed residues.</p>
<div id="fig-rama2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://proteopedia.org/wiki/index.php/Ramachandran_Plot"><img src="pics/Ramachandran_Gly_Pro_data_and_contours_T8000_small.jpg" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">Figure&nbsp;9: Ramachandran plots for glycine (lef) and proline (right) Inner contour encloses 98% and 99.9% of Top structures data, indicating the favored and allowed regions, respectively .</figcaption>
</figure>
</div>
</section>
<section id="protein-folds-domains-and-motifs" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="protein-folds-domains-and-motifs"><span class="header-section-number">2.2</span> Protein folds, domains and motifs</h2>
<p>The final three dimensional tertiary structure of a protein is commonly referred as its <strong>fold</strong>. Within the overall protein fold, we can recognize distinct <strong>domains</strong> and <strong>motifs.</strong> Domains are compact sections of the protein that represent structurally and (usually) functionally independent regions. That means that a domain maintain its main features, even if separated from the overall protein. On the other hand, motifs are small substructures that are not necessarily independent and consist of only a few secondary structure stretches. Indeed, motifs can be also referred as <em>super-secondary</em> structure.</p>
<p>The diversity of protein folds, domains and motifs, and combination of those, can be used for classification of protein structures hierarchically, as in many other fields of biology. The first classification was proposed in the 70’s and consisted of four groups of folds, as shown in the figure below. <em>All α</em> proteins are based almost entirely on an α-helical structure, and <em>all β</em>-structure are based on β-sheets. <em>α/β</em> structure is based on as mixture of α-helices and β-sheet, often organized as parallel β-strands connected by α-helices. On the other hand, <em>α+β</em> structures consist of discrete α-helix and β-sheet motifs that are not interwoven (as they are in α/β proteins). Finally, <em>small proteins</em> span polypeptides with no or little secondary structures.</p>
<div id="fig-chlothia" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pics/clasif.png" class="figure img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;10: The four structural protein classes in the classification by Chlothia &amp; Levitt. Modified from <span class="citation" data-cites="structur">Gu and Bourne (<a href="#ref-structur" role="doc-biblioref">2009</a>)</span> using 1I2T, 1K76, 1H75 and 1EM7 structures.</figcaption>
</figure>
</div>
</section>
<section id="quick-exercise" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="quick-exercise"><span class="header-section-number">2.3</span> <span style="color:green"><strong>Quick exercise</strong></span></h2>
<p>Explore the structure below and try to understand its structure. Then answer the question.</p>
    <div id="app-3MWD" class="molstar-app"></div>
    <script type="text/javascript">
    molstar.Viewer.create("app-3MWD", {"layoutShowSequence":false,"viewportShowExpand":true,"layoutIsExpanded":false,"layoutShowLog":false,"layoutShowLeftPanel":true,"layoutShowRemoteState":false,"layoutShowControls":false,"emdbProvider":"rcsb","pdbProvider":"rcsb","viewportShowAnimation":false,"viewportShowSelectionMode":false}).then(viewer => {
    viewer.loadPdb("3MWD");    });
    </script>
    
<strong>Which structural class is the protein above?</strong>
<div id="radio_KLKEEKWMSG" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_KLKEEKWMSG" value=""> <span>All-α</span></label><label><input type="radio" autocomplete="off" name="radio_KLKEEKWMSG" value=""> <span>All β</span></label><label><input type="radio" autocomplete="off" name="radio_KLKEEKWMSG" value="answer"> <span>α/β</span></label><label><input type="radio" autocomplete="off" name="radio_KLKEEKWMSG" value=""> <span>α+β</span></label><label><input type="radio" autocomplete="off" name="radio_KLKEEKWMSG" value=""> <span>Small</span></label>
</div>
<p>As you can see, as the protein is larger, classification gets more difficult. Moreover, as fold space has become more and more complex, these types of classifications have been adjusted and extended such that a complete hierarchy is created. The most commonly referred approaches to this sort of classification are those used by SCOP and CATH databases, as we will see in the <a href="ddbb.html#strDDBB">Structural Databases</a> section.</p>
</section>
<section id="hands-on-playing-with-secondary-structures" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="hands-on-playing-with-secondary-structures"><span class="header-section-number">2.4</span> <span style="color:green"><strong>Hands on: Playing with secondary structures</strong></span></h2>
<p><img src="pics/handson.png" title="Hands-on" class="img-fluid"></p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Groups
</div>
</div>
<div class="callout-body-container callout-body">
<p>Remember to work on groups as assigned. Groups should be the same for all the course exercises.</p>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Evernote
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>This is a <a href="https://evernote.com/intl/es">Evernote</a> note that you can check on your web browser and also copy into your Evenote account/App if you wish. Link <a href="https://www.evernote.com/shard/s62/nl/6844171/146913e2-a71b-a089-bcc5-c99ec0e5060b?title=StrBio%202023/2024%20-%201.%20(A)%20Secondary%20structures">here</a></p>
</div>
</div>
</div>
<p>There are a few online alternatives to model any peptide sequence and quickly see the effect of amino acid composition in the secondary structure. One of the best-known is Foldit (<a href="http://www.fold.it">www.fold.it</a>, <span class="citation" data-cites="miller2020">Miller et al. (<a href="#ref-miller2020" role="doc-biblioref">2020</a>)</span>), a gaming platform for biochemistry and structural biology teaching. It is a highly recommended alternative for most courses related to protein structure.</p>
<p>In this course we are going to try a more recent proposal, recently twitted by Sergey Ovchinnikov:</p>
<p></p><div id="tweet-58564"></div><script>tweet={"url":"https:\/\/twitter.com\/sokrypton\/status\/1535857255647690753","author_name":"Sergey Ovchinnikov \uD83C\uDDFA\uD83C\uDDE6","author_url":"https:\/\/twitter.com\/sokrypton","html":"\u003Cblockquote class=\"twitter-tweet\" align=\"center\"\u003E\u003Cp lang=\"en\" dir=\"ltr\"\u003EFor my latest attempt at introducing proteins to students, I made a Google Colab Notebook that predicts proteins from a single sequence. I asked the students to tweak the sequence to get a helix or two helices or... (1\/5)\u003Ca href=\"https:\/\/t.co\/eRufxDAkWm\"\u003Ehttps:\/\/t.co\/eRufxDAkWm\u003C\/a\u003E \u003Ca href=\"https:\/\/t.co\/KLkfrvetL8\"\u003Epic.twitter.com\/KLkfrvetL8\u003C\/a\u003E\u003C\/p\u003E&mdash; Sergey Ovchinnikov \uD83C\uDDFA\uD83C\uDDE6 (@sokrypton) \u003Ca href=\"https:\/\/twitter.com\/sokrypton\/status\/1535857255647690753?ref_src=twsrc%5Etfw\"\u003EJune 12, 2022\u003C\/a\u003E\u003C\/blockquote\u003E\n\u003Cscript async src=\"https:\/\/platform.twitter.com\/widgets.js\" charset=\"utf-8\"\u003E\u003C\/script\u003E\n","width":550,"height":null,"type":"rich","cache_age":"3153600000","provider_name":"Twitter","provider_url":"https:\/\/twitter.com","version":"1.0"};document.getElementById("tweet-58564").innerHTML = tweet["html"];</script><p></p>
<p>This is based on ColabFold (see <a href="https://github.com/sokrypton/ColabFold" class="uri">https://github.com/sokrypton/ColabFold</a> and <span class="citation" data-cites="mirdita2022">Mirdita et al. (<a href="#ref-mirdita2022" role="doc-biblioref">2022</a>)</span>), an Alphafold2 (see <span class="citation" data-cites="jumper2021">Jumper et al. (<a href="#ref-jumper2021" role="doc-biblioref">2021</a>)</span>) free notebook in <a href="https://colab.research.google.com/?hl=en">Google Colab notebook</a>. All you need is a Google account and the following <em>cheatsheet</em>.</p>
<div id="fig-single" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://twitter.com/sokrypton/status/1535857255647690753"><img src="pics/cheatsheet.png" title="Protein hallucination" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">Figure&nbsp;11: Basic protein amino acids stats for protein design with ColabFold Single</figcaption>
</figure>
</div>
<p>Now go to ColabFold Single: <a href="https://colab.research.google.com/github/sokrypton/af_backprop/blob/beta/examples/AlphaFold_single.ipynb" class="uri">https://colab.research.google.com/github/sokrypton/af_backprop/blob/beta/examples/AlphaFold_single.ipynb</a></p>
<p>Construct some small proteins and compare the output. Note that the first model will take 3-5 min, but the others will be faster. I provide here some interesting examples (IUPAC one-letter amino acid code):</p>
<ol type="1">
<li><p>AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA</p></li>
<li><p>KKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK</p></li>
<li><p>PVAVEARENGRLAVRVEGAIAVLIRENGRLVVRVEGG</p></li>
<li><p>PELEKHREELGEFLKKETGIAVEIRENGRLEVRVEGYTDVKIEGGTERLKRFLEEL</p></li>
<li><p>ACTWEGNKLTCA</p></li>
</ol>
<p><strong>1. Answer the following questions:</strong><br>
<strong>- Why is a poly-K more stable (dark blue) than a poly-A?</strong><br>
<br>
<strong>- Could you predict the structure of a poly-V or a poly-G?</strong><br>
<br>
<strong>- What would happen if you introduce a K5W in the structure number 2? and in the 4?</strong><br>
<br>
<br>
<strong>2. Now, try to create peptides with a custom motif, such as:</strong><br>
<br>
<strong>- Two helices.</strong><br>
<strong>- A four-strands beta-sheet.</strong><br>
<strong>- Alpha-beta-beta-alpha.</strong><br>
</p>



</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-callaway2020" class="csl-entry" role="listitem">
Callaway, Ewen. 2020. <span>“Revolutionary Cryo-EM Is Taking over Structural Biology.”</span> <em>Nature</em> 578 (7794): 201–1. <a href="https://doi.org/10.1038/d41586-020-00341-9">https://doi.org/10.1038/d41586-020-00341-9</a>.
</div>
<div id="ref-structur" class="csl-entry" role="listitem">
Gu, J., and P. E. Bourne. 2009. <em>Structural Bioinformatics, 2nd Edition | Wiley</em>. <a href="https://www.wiley.com/en-us/Structural+Bioinformatics%2C+2nd+Edition-p-9780470181058">https://www.wiley.com/en-us/Structural+Bioinformatics%2C+2nd+Edition-p-9780470181058</a>.
</div>
<div id="ref-harder2010" class="csl-entry" role="listitem">
Harder, Tim, Wouter Boomsma, Martin Paluszewski, Jes Frellsen, Kristoffer E. Johansson, and Thomas Hamelryck. 2010. <span>“Beyond rotamers: a generative, probabilistic model of side chains in proteins.”</span> <em>BMC bioinformatics</em> 11 (June): 306. <a href="https://doi.org/10.1186/1471-2105-11-306">https://doi.org/10.1186/1471-2105-11-306</a>.
</div>
<div id="ref-jumper2021" class="csl-entry" role="listitem">
Jumper, John, Richard Evans, Alexander Pritzel, Tim Green, Michael Figurnov, Olaf Ronneberger, Kathryn Tunyasuvunakool, et al. 2021. <span>“Highly Accurate Protein Structure Prediction with AlphaFold.”</span> <em>Nature</em> 596 (7873): 583–89. <a href="https://doi.org/10.1038/s41586-021-03819-2">https://doi.org/10.1038/s41586-021-03819-2</a>.
</div>
<div id="ref-miller2020" class="csl-entry" role="listitem">
Miller, Josh Aaron, Firas Khatib, Haley Hammond, Seth Cooper, and Scott Horowitz. 2020. <span>“Introducing Foldit Education Mode.”</span> <em>Nature Structural &amp; Molecular Biology</em> 27 (9): 769–70. <a href="https://doi.org/10.1038/s41594-020-0485-6">https://doi.org/10.1038/s41594-020-0485-6</a>.
</div>
<div id="ref-mirdita2022" class="csl-entry" role="listitem">
Mirdita, Milot, Konstantin Schütze, Yoshitaka Moriwaki, Lim Heo, Sergey Ovchinnikov, and Martin Steinegger. 2022. <span>“ColabFold: making protein folding accessible to all.”</span> <em>Nature Methods</em> 19 (6): 679–82. <a href="https://doi.org/10.1038/s41592-022-01488-1">https://doi.org/10.1038/s41592-022-01488-1</a>.
</div>
<div id="ref-paiva2022" class="csl-entry" role="listitem">
Paiva, Vinícius de Almeida, Isabela de Souza Gomes, Cleiton Rodrigues Monteiro, Murillo Ventura Mendonça, Pedro Magalhães Martins, Charles Abreu Santana, Valdete Gonçalves-Almeida, Sandro Carvalho Izidoro, Raquel Cardoso de Melo-Minardi, and Sabrina de Azevedo Silveira. 2022. <span>“Protein Structural Bioinformatics: An Overview.”</span> <em>Computers in Biology and Medicine</em> 147 (August): 105695. <a href="https://doi.org/10.1016/j.compbiomed.2022.105695">https://doi.org/10.1016/j.compbiomed.2022.105695</a>.
</div>
<div id="ref-stollar2020" class="csl-entry" role="listitem">
Stollar, Elliott J, and David P Smith. 2020. <span>“Uncovering Protein Structure.”</span> <em>Essays in Biochemistry</em> 64 (4): 649–80. <a href="https://doi.org/10.1042/EBC20190042">https://doi.org/10.1042/EBC20190042</a>.
</div>
</div></section></div></main> <!-- /main -->
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
<nav class="page-navigation column-body">
  <div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">StrBio 2023-24</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./ddbb.html" class="pagination-link">
        <span class="nav-page-text">Protein Databases</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">Modesto Redrejo Rodríguez, 2023</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://creativecommons.org/licenses/by-nc/2.0/">
      <i class="bi bi-badge-cc-fill" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://twitter.com/mredrejo"><iconify-icon inline="" icon="fa6-brands:x-twitter"></iconify-icon></a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link active" href="https://github.com/StrBio/strbio.github.io" aria-current="page">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



<script src="site_libs/quarto-html/zenscroll-min.js"></script>
<script src="https://platform.twitter.com/widgets.js"></script>
</body></html>