<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.56">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-08-21">

<title>Introduction – Structural Bioinformatics at UAM</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./ddbb.html" rel="next">
<link href="./index.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/iconify-1.0.0-beta.2/iconify-icon.min.js"></script>
<script src="site_libs/quarto-contrib/molstar-v3.13.0/molstar.js"></script>
<link href="site_libs/quarto-contrib/molstar-v3.13.0/molstar.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/molstar-v3.13.0/app-container.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>


<link rel="stylesheet" href="styles.css">
<link rel="stylesheet" href="include/webex.css">
<meta name="twitter:title" content="Introduction – Structural Bioinformatics at UAM">
<meta name="twitter:description" content="">
<meta name="twitter:card" content="summary">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./intro.html">Background</a></li><li class="breadcrumb-item"><a href="./intro.html">Introduction</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Structural Bioinformatics at UAM</a> 
        <div class="sidebar-tools-main">
    <a href="mailto:modesto.redrejo@uam.es" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-envelope-paper"></i></a>
    <a href="https://github.com/StrBio/strbio.github.io" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">StrBio UAM</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Background</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ddbb.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pdb.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein Structures</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Modeling</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./homology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Homology Modeling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1d.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1D Features</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./advanced.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Advanced homology modeling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ai.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein modeling in the AlphaFold era</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./links.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Modeling Resources</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul class="collapse">
  <li><a href="#goals" id="toc-goals" class="nav-link active" data-scroll-target="#goals"><span class="header-section-number">1</span> Goals</a></li>
  <li><a href="#sec-str" id="toc-sec-str" class="nav-link" data-scroll-target="#sec-str"><span class="header-section-number">2</span> Before going forward: Protein Structure 101</a>
  <ul class="collapse">
  <li><a href="#sec-rama" id="toc-sec-rama" class="nav-link" data-scroll-target="#sec-rama"><span class="header-section-number">2.1</span> Ramachandran Plot</a></li>
  <li><a href="#protein-folds-domains-and-motifs" id="toc-protein-folds-domains-and-motifs" class="nav-link" data-scroll-target="#protein-folds-domains-and-motifs"><span class="header-section-number">2.2</span> Protein folds, domains and motifs</a></li>
  <li><a href="#quick-exercise" id="toc-quick-exercise" class="nav-link" data-scroll-target="#quick-exercise"><span class="header-section-number">2.3</span> <span style="color:green"><strong>Quick exercise</strong></span></a></li>
  <li><a href="#practice-1-playing-with-secondary-structures" id="toc-practice-1-playing-with-secondary-structures" class="nav-link" data-scroll-target="#practice-1-playing-with-secondary-structures"><span class="header-section-number">2.4</span> <span style="color:green"><strong>Practice 1: Playing with secondary structures</strong></span></a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./intro.html">Background</a></li><li class="breadcrumb-item"><a href="./intro.html">Introduction</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Introduction</h1>
</div>



<div class="quarto-title-meta column-body">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 21, 2023</p>
    </div>
  </div>
  
    <div>
    <div class="quarto-title-meta-heading">Modified</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">January 29, 2025</p>
    </div>
  </div>
    
  </div>
  


</header>


<section id="goals" class="level1" data-link="Preface" data-number="1">
<h1 data-link="Preface" data-number="1"><span class="header-section-number">1</span> Goals</h1>
<p><a href="https://en.wikipedia.org/wiki/Structural_bioinformatics" title="Wikipedia">Structural Bioinformatics</a> (SB) is a broad discipline that encompasses data resources, algorithms, and tools for investigating, analyzing, predicting, and interpreting biomacromolecular structures <span class="citation" data-cites="paiva2022">(<a href="#ref-paiva2022" role="doc-biblioref">Paiva et al. 2022</a>)</span>. In this course, we will specifically focus on protein structural bioinformatics, including visualization and analysis of the structure of biomacromolecules as well as the prediction of protein structures and complexes. The premise of SB is that high-resolution structural information about biological systems enables precise reasoning about their functions and the effects of modifications and perturbations.</p>
<p>The goals of SB require at least four different research lines (see Chapter 1 in <span class="citation" data-cites="structur">Gu and Bourne (<a href="#ref-structur" role="doc-biblioref">2009</a>)</span>):</p>
<ol type="1">
<li><p><em>Visualization</em>: Dealing with one or many complex structures and integrating various sources of information such as sequences, structural data, electrostatic fields, locations of functional sites, and areas of variability.</p></li>
<li><p><em>Classification</em>: Grouping similar structures hierarchically to identify common origins and diversification paths. Similar to other fields of biology classification is tedious but required to understand the structural space.</p></li>
<li><p><em>Prediction</em> of structures remains an area of keen interest and a field of research itself. As we will see below, the number of different sequences is much higher than the availability of structures, which make prediction an essential and useful tool.</p></li>
<li><p><em>Simulation.</em> Experimentally obtained structures are primarily static structural models (see warning below). However, the properties of these molecules are often the results of their dynamic motions. The definition of energy functions that govern the folding of proteins and their subsequent stable dynamics can be analyzed by molecular dynamics simulations, although computation capacities may be limiting to reach a biologically relevant timescales.</p></li>
</ol>
<p>Powered by vast amounts of data and significant technical advances, the field has undergone a substantial transformation over the past twenty years The enhancement of experimental capabilities to analyze the structure of proteins and other biological molecules and structures (see <span class="citation" data-cites="callaway2020">Callaway (<a href="#ref-callaway2020" role="doc-biblioref">2020</a>)</span>) and the advancement of Artificial Intelligence (AI)-assisted structure prediction have substantially increased the ability of life-science researchers to address various questions concerning protein diversity, evolution, and function. This transformation has boosted in the last 5 years, and its implications for biology, biotechnology, and biomedicine remain largely unpredictable.</p>
</section>
<section id="sec-str" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Before going forward: Protein Structure 101</h1>
<p>While it is possible to perform some protein modeling without expertise in structural biology, having a basic understanding of protein structure is highly recommended. In this course, there are often students who do not have a strong background in biology. Additionally, it has been observed that graduate students in biology, biomedicine, and related fields possess varying levels of knowledge about protein structure. For those looking to review or update their understanding of protein structure, it is suggested to read Chapter 2 of <span class="citation" data-cites="structur">Gu and Bourne (<a href="#ref-structur" role="doc-biblioref">2009</a>)</span>, the great review by <span class="citation" data-cites="stollar2020">Stollar and Smith (<a href="#ref-stollar2020" role="doc-biblioref">2020</a>)</span> and the <a href="https://en.wikipedia.org/wiki/Protein_structure">Wikipedia</a> and <a href="https://proteopedia.org/wiki/index.php/Introduction_to_protein_structure">Proteopedia</a> articles on protein structures, which were the primary sources for this brief section (follow picture links).</p>
<div id="fig-str" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-str-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="https://en.wikipedia.org/wiki/Protein_structure"><img src="pics/Protein_structure.png" title="Protein Structure" class="figure img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-str-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Protein structure levels, using human PCNA (PDB 1AXC) as an example.
</figcaption>
</figure>
</div>
<p>Proteins are essential components of life, involved in various vital functions as structural elements, scaffolding elements, or active enzymes that catalyze metabolic reactions. Proteins are composed of polymers of amino acids, and the sequence of amino acids of a particular protein is referred to as the <strong>Primary structure</strong> of the protein. Amino acid chains can spontaneously fold into three-dimensional structures, mainly stabilized by hydrogen bonds between amino acids. The amino acid sequence determines different layers of 3D structure. Each of the 20 natural amino acids has specific physicochemical properties that influence its preferred conformation. Therefore, the initial level of folding is known as the <strong>Secondary structure</strong>, which forms common patterns as will be discussed further. These segments of secondary structure patterns are capable of folding into three-dimensional forms due to interactions between the side chains of amino acids, known as protein <strong>Tertiary structure</strong>. Additionally, two or more individual peptide chains can aggregate to form multisubunit proteins, referred to as having <strong>Quaternary structure</strong>.</p>
<div id="fig-aa" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-aa-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="https://www.reddit.com/r/chemistry/comments/acyald/venn_diagram_showing_the_properties_of_the_20/"><img src="pics/aa.png" title="Amino acids clasification by type" class="figure img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-aa-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Amino acids classification by type
</figcaption>
</figure>
</div>
<p>It is important to note that the peptide bond itself does not permit rotation as it possesses partial double bond characteristics. Hence, rotation is restricted to the bonds between the <code>Cα</code> and the <code>C = O</code> group (the phi (φ) angle) and the <code>Cα</code> and the <code>NH</code> group (the psi (ψ) angle). The polypeptide backbone chain thus consists of a repeating sequence of two rotatable bonds followed by one non-rotatable (peptide) bond. However, not all 360º of the φ and ψ angles are feasible due to potential steric clashes between neighboring sidechains. For specific angles and amino acid combinations, spatial constraints prevent atoms from occupying the same physical location, which partially accounts for the varying propensities of certain amino acids to adopt different types of secondary structures.</p>
<div id="fig-bond" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-bond-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="https://portlandpress.com/essaysbiochem/article/64/4/649/226515/Uncovering-protein-structure"><img src="pics/peptide_bond.png" title="Peptide bond" class="figure img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-bond-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Scheme of a generic polypeptide chain. Residue side chains are denoted as R. Coloured rectangles indicate sets of six atoms that are coplanar due to the double-bond character of the peptide bond. Arrows indicate the bonds that are free to rotate with the angle of rotation about the N–Cα known as phi (φ) and about the Cα–C known as psi (ψ). Note that only peptide backbone bonds are labeled and in most cases the R group bond is free to rotate.
</figcaption>
</figure>
</div>
<p>Furthermore, the side chains of amino acids possess their own torsion angles, known as χ1, χ2, χ3, and so forth (<a href="#fig-chi" class="quarto-xref">Figure&nbsp;6</a>). These torsion angles significantly influence the secondary and, particularly, the tertiary structures of proteins. The various combinations of side-chain torsions defined by χ angles are referred to as <em>rotamers</em>.</p>
<p>Within these constraints, the two primary local conformations that avoid steric hindrance and maximize backbone-backbone hydrogen bonding are the α-helix and the β-sheet secondary structures. <a href="https://en.wikipedia.org/wiki/Linus_Pauling">Linus Pauling</a> initially proposed the <strong>α-helix</strong> as left-handed in 1951, but the myoglobin crystal structure in 1958 revealed that the right-handed form is more common. In the typical right-handed helices, the backbone NH group hydrogen bonds to the backbone C=O group of the amino acid located four residues earlier along the protein sequence. This regular coil shape has the R-groups pointing outward, away from the peptide backbone, and requires about 3.6 residues to complete a full turn of the helix (<a href="#fig-alpha" class="quarto-xref">Figure&nbsp;4</a>).</p>
<div>

</div>
<div class="quarto-layout-panel" data-layout-ncol="1">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-alpha" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-alpha-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="https://en.wikipedia.org/wiki/Alpha_helix"><img src="pics/alpha.png" title="Alpha helix" class="figure img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-alpha-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Alpha helix.
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-beta" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-beta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="https://en.wikipedia.org/wiki/Beta_sheet"><img src="pics/beta.png" title="Beta sheet" class="figure img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-beta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Detailed description of a beta sheet made up of three beta strands.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<p>Different amino-acid sequences have varying tendencies to form α-helical structures. <a href="https://en.wikipedia.org/wiki/Methionine" title="Methionine">Methionine</a>, <a href="https://en.wikipedia.org/wiki/Alanine" title="Alanine">alanine</a>, <a href="https://en.wikipedia.org/wiki/Leucine" title="Leucine">leucine</a>, <a href="https://en.wikipedia.org/wiki/Glutamate" title="Glutamate">glutamate</a>, and <a href="https://en.wikipedia.org/wiki/Lysine" title="Lysine">lysine</a> have especially high helix-forming propensities, whereas <a href="https://en.wikipedia.org/wiki/Proline" title="Proline">proline</a> and <a href="https://en.wikipedia.org/wiki/Glycine" title="Glycine">glycine</a> have poor helix-forming propensities. <a href="https://en.wikipedia.org/wiki/Proline" title="Proline">Proline</a> often disrupts or kinks a helix because it lacks an amide hydrogen to form <a href="https://en.wikipedia.org/wiki/Hydrogen_bond" title="Hydrogen bond">hydrogen bonds</a> and its bulky side chain interferes with the preceding turn’s backbone. <a href="https://en.wikipedia.org/wiki/Glycine" title="Glycine">Glycine</a> with only a hydrogen as its R-group, is too flexible and entropically costly to maintain the constrained α-helical structure, making it an α-helix breaker.</p>
<p><strong>β-Sheets</strong> (<a href="#fig-beta" class="quarto-xref">Figure&nbsp;5</a>) consist of two or more extended polypeptide chains called β-strands running alongside each other in either a parallel or antiparallel arrangement. In a β-sheet, residues arrange themselves in a zigzag pattern with adjacent peptide bonds pointing in opposite directions. The NH group and the C=O group of each amino acid form hydrogen bonds with the C=O group and NH group, respectively, on adjacent strands. The chains can run in opposite directions (<em>antiparallel β-sheet</em>) or the same direction (<em>parallel β-sheet</em>). Sidechains from each residue alternate in opposite directions, giving β-sheets hydrophilic and hydrophobic faces, often forming a pattern of alternating hydrophilic and hydrophobic residues in the primary structure.</p>
<p>Large aromatic residues (<a href="https://en.wikipedia.org/wiki/Tyrosine" title="Tyrosine">tyrosine</a>, <a href="https://en.wikipedia.org/wiki/Phenylalanine" title="Phenylalanine">phenylalanine</a>, <a href="https://en.wikipedia.org/wiki/Tryptophan" title="Tryptophan">tryptophan</a>) and β-branched amino acids (<a href="https://en.wikipedia.org/wiki/Threonine" title="Threonine">threonine</a>, <a href="https://en.wikipedia.org/wiki/Valine" title="Valine">valine</a>, <a href="https://en.wikipedia.org/wiki/Isoleucine" title="Isoleucine">isoleucine</a>) are commonly found in β-strands. As in the case of α-helixes, β-strands are often ended by <a href="https://en.wikipedia.org/wiki/Glycine" title="Glycine">glycines</a>, which are especially common in β-turns (the most common connector between strands), as amino acids with positive φ angles.</p>
<div id="fig-chi" class="figure quarto-float quarto-figure quarto-figure-center anchored" width="301">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-chi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-306"><img src="images/paste-6EE800AE.png" class="figure img-fluid figure-img" width="301"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-chi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Dihedral angles in glutamate: Dihedral angles are the main degrees of freedom for the backbone (ϕ and ψ angles) and the side chain (χ angles) of an amino acid. The number of χ angles varies between zero and four for the 20 standard amino acids. The figure shows a ball-and-stick representation of glutamate, which has three χ angles.
</figcaption>
</figure>
</div>
<p>Protein secondary, tertiary, and quaternary structures are maintained by interactions between amino acids (<a href="#fig-interaction" class="quarto-xref">Figure&nbsp;7</a>). These interactions are typically classified into four types:</p>
<ol type="1">
<li><em>Ionic bonding</em>: Ionic bonds arise from electrostatic attractions between positively and negatively charged amino acid side chains. For example, the attraction between an aspartic acid carboxylate ion and a lysine ammonium ion helps stabilize a specific folded region of a protein.</li>
<li><em>Hydrogen bonding</em>: Hydrogen bonds form between a highly electronegative oxygen or nitrogen atom and a hydrogen atom attached to another oxygen or nitrogen atom, such as those in polar amino acid side chains. Hydrogen bonds are crucial for both intra- and intermolecular interactions in proteins, like in alpha-helixes.</li>
<li><em>Disulfide linkages</em>. When two cysteine amino acids come close together during protein folding under appropriate redox conditions, oxidation can link their sulfur atoms, forming a disulfide bond. Unlike ionic or hydrogen bonds, these are covalent links, and are thus a classic example of a spontaneus reaction, occurring as a post-translational modification. Although they are sensitive to reducing agents, greatly stabilize the tertiary structure and are vital for the quaternary structure of many proteins, like antibodies.</li>
<li><em>Hydrophobic interactions.</em> Dispersion forces arise when a normally nonpolar atom becomes momentarily polar due to an uneven distribution of electrons, leading to an instantaneous dipole that induces a shift of electrons in a neighboring nonpolar atom. Dispersion forces are weak but can be important when other types of interactions are either missing or minimal. The term <em>hydrophobic interaction</em> is often misused as a synonym for dispersion forces. Hydrophobic interactions arise because water molecules engage in hydrogen bonding with other water molecules (or groups in proteins capable of hydrogen bonding). Because nonpolar groups cannot engage in hydrogen bonding, the protein folds in such a way that these groups are buried in the interior part of the protein structure, minimizing their contact with water.</li>
</ol>
<div id="fig-interaction" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-interaction-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/interactions-01.png" class="figure img-fluid figure-img" width="574">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-interaction-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: Four interactions stabilize the tertiary structure of a protein: (a) ionic bonding, (b) hydrogen bonding, (c) disulfide linkages, and (d) dispersion forces
</figcaption>
</figure>
</div>
<p>Other less frequent intramolecular interactions might be relevant in some proteins, like the so-called <strong>Isopeptide bonds</strong>, formed between two protein groups, at least one of which is not an α-amino or α-carboxy group. Examples include ubiquitylation, sumoylation, transglutamination, sortase-mediated cell surface protein anchoring and pilus formation <span class="citation" data-cites="kang2011">(<a href="#ref-kang2011" role="doc-biblioref">Kang and Baker 2011</a>)</span>. As detailed in <a href="#fig-iso" class="quarto-xref">Figure&nbsp;8</a>, all these processes share several feature:</p>
<ol type="1">
<li>All involve the reaction of a lysine ε-amino group on one protein with the main α-carboxy group on another protein, except for transglutamination, where the lysine targets a glutamine side chain carboxyamide group.</li>
<li>All processes are enzyme-mediated and involve a transient thioester intermediate formed by the active site’s cysteine. This intermediate is resolved through nucleophilic attack by the lysine ε-amino group, which completes the bond formation.</li>
</ol>
<div id="fig-iso" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-iso-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="https://www.sciencedirect.com/science/article/pii/S0968000410001842"><img src="images/iso.jpg" class="figure img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-iso-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: Enzyme-mediated formation of intermolecular isopeptide bonds. Examples are shown for three different biological processes: ubiquitylation, transglutamination and sortase-mediated pilus assembly in Gram-positive bacteria. Proteins linked by isopeptide bonds are coloured blue and purple and the isopeptide bond-forming enzymes are coloured green. Reproduced from <span class="citation" data-cites="kang2011">Kang and Baker (<a href="#ref-kang2011" role="doc-biblioref">2011</a>)</span>.
</figcaption>
</figure>
</div>
<p>In contrast to these enzyme-dependent processes, cross-linking isopeptide bonds form autocatalytically in <em>S. pyogenes</em> major pilin Spy0128 and other Gram+ cell surface proteins, as well as in HK97 phage capsid. In this case, the bond-forming reaction is a proximity-induced reaction that happens when participating amino acids are positioned together in a hydrophobic environment, either through protein folding concurrent with peptide bond formation at the ribosome or by capsid rearrangement (in HK97).</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Unnatural reactive amino acids
</div>
</div>
<div class="callout-body-container callout-body">
<p>Manipulating inter- and intramolecular amino acid interactions is common for altering protein properties. A classic example is adding nearby cysteines to form disulfide bonds, enhancing stability. Recently, Spy0128 motifs and unnatural reactive amino acids have enabled applications like creating protein staples, thermostable antibodies, and surfaces/nanoparticles that capture proteins <span class="citation" data-cites="cao2022">(<a href="#ref-cao2022" role="doc-biblioref">Cao and Wang 2022</a>)</span>.</p>
</div>
</div>
<section id="sec-rama" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="sec-rama"><span class="header-section-number">2.1</span> Ramachandran Plot</h2>
<p>As you likely deduced already, many combinations of φ and ψ angles are prohibited due to the principle of steric exclusion, which dictates that two atoms cannot occupy the same space simultaneously. This concept was initially demonstrated by <a href="https://en.wikipedia.org/wiki/G._N._Ramachandran">Gopalasamudram Ramachandran</a>, who developed a plot to visualize the permissible angle values, known as the Ramachandran plot. This plot can display the angles of a specific amino acid, all amino acids in a single protein, or even across many proteins. Analysis of φ and ψ angles in known proteins reveals that approximately three-quarters of all possible φ, ψ combinations are not allowed (<a href="#fig-rama0" class="quarto-xref">Figure&nbsp;9</a>) that correspond with common secondary structure motifs (<a href="#fig-ram" class="quarto-xref">Figure&nbsp;10</a>).</p>
<div id="fig-rama0" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rama0-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="https://proteopedia.org/wiki/index.php/Ramachandran_Plot"><img src="pics/Ramachandran_plot_general_100K.jpg" title="General Ramachandran plot" class="figure img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rama0-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: General Ramachandran plot. The density of points reflects how likely is each angle combination, defining the core (red line) and tolerance (orange) regions.
</figcaption>
</figure>
</div>
<div id="fig-ram" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ram-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="pics/rama2.png" title="Textbook Rama plot" class="figure img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ram-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: Definition of secondary structure alternatives by their combination of phi, psi angles.
</figcaption>
</figure>
</div>
<p>Functionally relevant residues are more likely than others to have torsion angles that plot to the allowed but disfavored regions of a Ramachandran plot. The specific geometry of these functionally relevant residues, while somewhat energetically unfavorable, may be important for the protein’s function, catalytic or otherwise. Such conformations need to be stabilized by the protein using H-bonds, steric packing, or other means, and should very seldom occur for highly solvent-exposed residues.</p>
<div id="fig-rama2" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rama2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="https://proteopedia.org/wiki/index.php/Ramachandran_Plot"><img src="pics/Ramachandran_Gly_Pro_data_and_contours_T8000_small.jpg" class="figure img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rama2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;11: Ramachandran plots for glycine (left) and proline (right) Inner contour encloses 98% and 99.9% of Top structures data, indicating the favored and allowed regions, respectively .
</figcaption>
</figure>
</div>
</section>
<section id="protein-folds-domains-and-motifs" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="protein-folds-domains-and-motifs"><span class="header-section-number">2.2</span> Protein folds, domains and motifs</h2>
<p>The global three dimensional tertiary structure of a protein is commonly referred as its <strong>fold</strong>. Within the overall protein fold, we can recognize distinct <strong>domains</strong> and <strong>motifs.</strong> Domains are compact sections of the protein that represent structurally and (usually) functionally independent regions. That means that a domain maintain its main features, even if separated from the overall protein. On the other hand, motifs are small substructures that are not necessarily independent and consist of only a few secondary structure stretches. Indeed, motifs can be also referred as <em>super-secondary</em> structure.</p>
<p>The diversity of protein folds, domains and motifs, and combination of those, can be used for classification of protein structures hierarchically, as in many other fields of biology. The first classification was proposed in the 70’s and consisted of four groups of folds, as shown in the figure below. <em>All α</em> proteins are based almost entirely on an α-helical structure, and <em>all β</em>-structure are based on β-sheets. <em>α/β</em> structure is based on as mixture of α-helices and β-sheet, often organized as parallel β-strands connected by α-helices. On the other hand, <em>α+β</em> structures consist of discrete α-helix and β-sheet motifs that are not interwoven (as they are in α/β proteins). Finally, <em>small proteins</em> span polypeptides with no or little secondary structures.</p>
<div id="fig-chlothia" class="figure quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-chlothia-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="pics/clasif.png" class="figure img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-chlothia-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12: The four structural protein classes in the classification by Chlothia &amp; Levitt. Modified from <span class="citation" data-cites="structur">Gu and Bourne (<a href="#ref-structur" role="doc-biblioref">2009</a>)</span> using 1I2T, 1K76, 1H75 and 1EM7 structures.
</figcaption>
</figure>
</div>
</section>
<section id="quick-exercise" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="quick-exercise"><span class="header-section-number">2.3</span> <span style="color:green"><strong>Quick exercise</strong></span></h2>
<p>Explore the structure below and try to understand its structure. Then answer the question.</p>
    <div id="app-3MWD" class="molstar-app"></div>
    <script type="text/javascript">
    molstar.Viewer.create("app-3MWD", {"layoutShowSequence":false,"layoutShowControls":false,"layoutShowRemoteState":false,"viewportShowSelectionMode":false,"viewportShowAnimation":false,"pdbProvider":"rcsb","layoutIsExpanded":false,"emdbProvider":"rcsb","layoutShowLeftPanel":true,"layoutShowLog":false,"viewportShowExpand":true}).then(viewer => {
    viewer.loadPdb("3MWD");    });
    </script>
    
<strong>How many unique protein chains are there is this protein?</strong>
<div id="radio_BBJNQFAVWT" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_BBJNQFAVWT" value=""> <span>1</span></label><label><input type="radio" autocomplete="off" name="radio_BBJNQFAVWT" value="answer"> <span>2</span></label><label><input type="radio" autocomplete="off" name="radio_BBJNQFAVWT" value=""> <span>3</span></label><label><input type="radio" autocomplete="off" name="radio_BBJNQFAVWT" value=""> <span>&gt;3</span></label>
</div>
<strong>Which structural class is the protein above?</strong>
<div id="radio_QSLRRYFZXC" class="webex-radiogroup">
<label><input type="radio" autocomplete="off" name="radio_QSLRRYFZXC" value=""> <span>All-α</span></label><label><input type="radio" autocomplete="off" name="radio_QSLRRYFZXC" value=""> <span>All β</span></label><label><input type="radio" autocomplete="off" name="radio_QSLRRYFZXC" value="answer"> <span>α/β</span></label><label><input type="radio" autocomplete="off" name="radio_QSLRRYFZXC" value=""> <span>α+β</span></label><label><input type="radio" autocomplete="off" name="radio_QSLRRYFZXC" value=""> <span>Small</span></label>
</div>
<p>As you can see, as the protein is larger, classification gets more difficult. Moreover, as fold space has become more and more complex, these types of classifications have been adjusted and extended such that a complete hierarchy is created. The most commonly referred approaches to this sort of classification are those used by SCOP and CATH databases, as we will see in the <a href="ddbb.html#strDDBB">Structural Databases</a> section.</p>
</section>
<section id="practice-1-playing-with-secondary-structures" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="practice-1-playing-with-secondary-structures"><span class="header-section-number">2.4</span> <span style="color:green"><strong>Practice 1: Playing with secondary structures</strong></span></h2>
<p><img src="pics/handson.png" title="Hands-on" class="img-fluid"></p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Groups
</div>
</div>
<div class="callout-body-container callout-body">
<p>Remember to work on groups as assigned. Groups should be the same for all the course exercises.</p>
</div>
</div>
<p>There are a few online alternatives to model any peptide sequence and quickly see the effect of amino acid composition in the secondary structure. One of the best-known is Foldit (<a href="http://www.fold.it">www.fold.it</a>, <span class="citation" data-cites="miller2020">Miller et al. (<a href="#ref-miller2020" role="doc-biblioref">2020</a>)</span>), a gaming platform for biochemistry and structural biology teaching. It is a highly recommended alternative for most courses related to protein structure.</p>
<p>In this course we are going to try a more recent proposal, twitted by Sergey Ovchinnikov:</p>
<p></p><div id="tweet-18065"></div><script>tweet={"url":"https:\/\/twitter.com\/sokrypton\/status\/1535857255647690753","author_name":"Sergey Ovchinnikov","author_url":"https:\/\/twitter.com\/sokrypton","html":"\u003Cblockquote class=\"twitter-tweet\" align=\"center\"\u003E\u003Cp lang=\"en\" dir=\"ltr\"\u003EFor my latest attempt at introducing proteins to students, I made a Google Colab Notebook that predicts proteins from a single sequence. I asked the students to tweak the sequence to get a helix or two helices or... (1\/5)\u003Ca href=\"https:\/\/t.co\/eRufxDAkWm\"\u003Ehttps:\/\/t.co\/eRufxDAkWm\u003C\/a\u003E \u003Ca href=\"https:\/\/t.co\/KLkfrvetL8\"\u003Epic.twitter.com\/KLkfrvetL8\u003C\/a\u003E\u003C\/p\u003E&mdash; Sergey Ovchinnikov (@sokrypton) \u003Ca href=\"https:\/\/twitter.com\/sokrypton\/status\/1535857255647690753?ref_src=twsrc%5Etfw\"\u003EJune 12, 2022\u003C\/a\u003E\u003C\/blockquote\u003E\n\u003Cscript async src=\"https:\/\/platform.twitter.com\/widgets.js\" charset=\"utf-8\"\u003E\u003C\/script\u003E\n\n","width":550,"height":null,"type":"rich","cache_age":"3153600000","provider_name":"Twitter","provider_url":"https:\/\/twitter.com","version":"1.0"};document.getElementById("tweet-18065").innerHTML = tweet["html"];</script><p></p>
<div id="fig-single" class="figure quarto-float quarto-figure quarto-figure-center anchored" width="500">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-single-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="https://twitter.com/sokrypton/status/1535857255647690753"><img src="pics/cheatsheet.png" title="Protein hallucination" class="figure img-fluid figure-img" width="500"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-single-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13: Basic protein amino acids stats for protein design with ColabFold Single
</figcaption>
</figure>
</div>
<p>This is based on ColabFold (see <a href="https://github.com/sokrypton/ColabFold" class="uri">https://github.com/sokrypton/ColabFold</a> and <span class="citation" data-cites="mirdita2022">Mirdita et al. (<a href="#ref-mirdita2022" role="doc-biblioref">2022</a>)</span>), an Alphafold2 (see <span class="citation" data-cites="jumper2021">Jumper et al. (<a href="#ref-jumper2021" role="doc-biblioref">2021</a>)</span>) free notebook in <a href="https://colab.research.google.com/?hl=en">Google Colab notebook</a>. All you need is a Google account and the <em>cheatsheet</em> in <a href="#fig-single" class="quarto-xref">Figure&nbsp;13</a>.</p>
<p>Now go to ColabFold Single: <a href="https://colab.research.google.com/github/sokrypton/af_backprop/blob/beta/examples/AlphaFold_single.ipynb" class="uri">https://colab.research.google.com/github/sokrypton/af_backprop/blob/beta/examples/AlphaFold_single.ipynb</a></p>
<p>Construct some small proteins and compare the output. Use 6 recycle steps to obtain diverse structures and then click in the “Animate trajectories” chunk. Note that the first model will take 2-4 min, but the others will be faster. I provide here some interesting examples (IUPAC one-letter amino acid code):</p>
<ol type="1">
<li><p>AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA</p></li>
<li><p>KKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK</p></li>
<li><p>PVAVEARENGRLAVRVEGAIAVLIRENGRLVVRVEGG</p></li>
<li><p>PELEKHREELGEFLKKETGIAVEIRENGRLEVRVEGYTDVKIEGGTERLKRFLEEL</p></li>
<li><p>ACTWEGNKLTCA</p></li>
</ol>
<p><strong>1. Answer the following questions:</strong><br>
<strong>- Why is a poly-K more stable (dark blue) than a poly-A?</strong><br>
<br>
<strong>- Could you predict the structure of a poly-V or a poly-G?</strong><br>
<br>
<strong>- What would happen if you introduce a K5W in the structure number 2? and in the 4?</strong><br>
<br>
<br>
<strong>2. Now, try to create peptides with a custom motif, such as:</strong><br>
<br>
<strong>- Two helices.</strong><br>
<strong>- A four-strands beta-sheet.</strong><br>
<strong>- Alpha-beta-beta-alpha.</strong><br>
</p>



</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-callaway2020" class="csl-entry" role="listitem">
Callaway, Ewen. 2020. <span>“Revolutionary Cryo-EM Is Taking over Structural Biology.”</span> <em>Nature</em> 578 (7794): 201–1. <a href="https://doi.org/10.1038/d41586-020-00341-9">https://doi.org/10.1038/d41586-020-00341-9</a>.
</div>
<div id="ref-cao2022" class="csl-entry" role="listitem">
Cao, Li, and Lei Wang. 2022. <span>“New covalent bonding ability for proteins.”</span> <em>Protein Science: A Publication of the Protein Society</em> 31 (2): 312–22. <a href="https://doi.org/10.1002/pro.4228">https://doi.org/10.1002/pro.4228</a>.
</div>
<div id="ref-structur" class="csl-entry" role="listitem">
Gu, J., and P. E. Bourne. 2009. <em>Structural Bioinformatics, 2nd Edition | Wiley</em>. <a href="https://www.wiley.com/en-us/Structural+Bioinformatics%2C+2nd+Edition-p-9780470181058">https://www.wiley.com/en-us/Structural+Bioinformatics%2C+2nd+Edition-p-9780470181058</a>.
</div>
<div id="ref-jumper2021" class="csl-entry" role="listitem">
Jumper, John, Richard Evans, Alexander Pritzel, Tim Green, Michael Figurnov, Olaf Ronneberger, Kathryn Tunyasuvunakool, et al. 2021. <span>“Highly Accurate Protein Structure Prediction with AlphaFold.”</span> <em>Nature</em> 596 (7873): 583–89. <a href="https://doi.org/10.1038/s41586-021-03819-2">https://doi.org/10.1038/s41586-021-03819-2</a>.
</div>
<div id="ref-kang2011" class="csl-entry" role="listitem">
Kang, Hae Joo, and Edward N. Baker. 2011. <span>“Intramolecular isopeptide bonds: protein crosslinks built for stress?”</span> <em>Trends in Biochemical Sciences</em> 36 (4): 229–37. <a href="https://doi.org/10.1016/j.tibs.2010.09.007">https://doi.org/10.1016/j.tibs.2010.09.007</a>.
</div>
<div id="ref-miller2020" class="csl-entry" role="listitem">
Miller, Josh Aaron, Firas Khatib, Haley Hammond, Seth Cooper, and Scott Horowitz. 2020. <span>“Introducing Foldit Education Mode.”</span> <em>Nature Structural &amp; Molecular Biology</em> 27 (9): 769–70. <a href="https://doi.org/10.1038/s41594-020-0485-6">https://doi.org/10.1038/s41594-020-0485-6</a>.
</div>
<div id="ref-mirdita2022" class="csl-entry" role="listitem">
Mirdita, Milot, Konstantin Schütze, Yoshitaka Moriwaki, Lim Heo, Sergey Ovchinnikov, and Martin Steinegger. 2022. <span>“ColabFold: making protein folding accessible to all.”</span> <em>Nature Methods</em> 19 (6): 679–82. <a href="https://doi.org/10.1038/s41592-022-01488-1">https://doi.org/10.1038/s41592-022-01488-1</a>.
</div>
<div id="ref-paiva2022" class="csl-entry" role="listitem">
Paiva, Vinícius de Almeida, Isabela de Souza Gomes, Cleiton Rodrigues Monteiro, Murillo Ventura Mendonça, Pedro Magalhães Martins, Charles Abreu Santana, Valdete Gonçalves-Almeida, Sandro Carvalho Izidoro, Raquel Cardoso de Melo-Minardi, and Sabrina de Azevedo Silveira. 2022. <span>“Protein Structural Bioinformatics: An Overview.”</span> <em>Computers in Biology and Medicine</em> 147 (August): 105695. <a href="https://doi.org/10.1016/j.compbiomed.2022.105695">https://doi.org/10.1016/j.compbiomed.2022.105695</a>.
</div>
<div id="ref-stollar2020" class="csl-entry" role="listitem">
Stollar, Elliott J, and David P Smith. 2020. <span>“Uncovering Protein Structure.”</span> <em>Essays in Biochemistry</em> 64 (4): 649–80. <a href="https://doi.org/10.1042/EBC20190042">https://doi.org/10.1042/EBC20190042</a>.
</div>
</div></section></div></main> <!-- /main -->
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
          // target, if specified
          link.setAttribute("target", "_blank");
          if (link.getAttribute("rel") === null) {
            link.setAttribute("rel", "noopener");
          }
          // default icon
          link.classList.add("external");
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation column-body">
  <div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link" aria-label="StrBio UAM">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">StrBio UAM</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./ddbb.html" class="pagination-link" aria-label="Protein Databases">
        <span class="nav-page-text">Protein Databases</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>Modesto Redrejo Rodríguez, 2025</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://creativecommons.org/licenses/by-nc/2.0/">
      <i class="bi bi-badge-cc-fill" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://twitter.com/mredrejo">
<p><iconify-icon inline="" icon="fa6-brands:x-twitter"></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://bsky.app/profile/mredrejo.bsky.social">
<p><iconify-icon inline="" icon="simple-icons:bluesky"></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link active" href="https://github.com/StrBio/strbio.github.io" aria-current="page">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>




<script src="site_libs/quarto-html/zenscroll-min.js"></script>
<script src="https://platform.twitter.com/widgets.js"></script>
</body></html>