<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-08-21">

<title>Structural Bioinformatics at UAM - Modeling Resources</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./ai.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/iconify-1.0.0-beta.2/iconify-icon.min.js"></script>
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>


<link rel="stylesheet" href="styles.css">
<link rel="stylesheet" href="include/webex.css">
<meta name="twitter:title" content="Structural Bioinformatics at UAM - Modeling Resources">
<meta name="twitter:card" content="summary">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./homology.html">Modeling</a></li><li class="breadcrumb-item"><a href="./links.html">Modeling Resources</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Structural Bioinformatics at UAM</a> 
        <div class="sidebar-tools-main">
    <a href="mailto:modesto.redrejo@uam.es" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-envelope-paper"></i></a>
    <a href="https://github.com/StrBio/strbio.github.io" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">StrBio 2023-24</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Background</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ddbb.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pdb.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein Structures</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Modeling</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./homology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Homology Modeling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1d.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1D Features</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./advanced.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Advanced homology modeling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ai.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein modeling in the AlphaFold era</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./links.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Modeling Resources</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul class="collapse">
  <li><a href="#template-search" id="toc-template-search" class="nav-link active" data-scroll-target="#template-search"><span class="header-section-number">1</span> <strong>Template search</strong></a></li>
  <li><a href="#secondary-structure-and-1d-features-services" id="toc-secondary-structure-and-1d-features-services" class="nav-link" data-scroll-target="#secondary-structure-and-1d-features-services"><span class="header-section-number">2</span> <strong>Secondary structure and 1D features services</strong></a></li>
  <li><a href="#protein-modeling" id="toc-protein-modeling" class="nav-link" data-scroll-target="#protein-modeling"><span class="header-section-number">3</span> <strong>Protein modeling</strong></a></li>
  <li><a href="#model-quality-assessment" id="toc-model-quality-assessment" class="nav-link" data-scroll-target="#model-quality-assessment"><span class="header-section-number">4</span> <strong>Model quality assessment</strong></a></li>
  <li><a href="#other-resources" id="toc-other-resources" class="nav-link" data-scroll-target="#other-resources"><span class="header-section-number">5</span> <strong>Other resources</strong></a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Modeling Resources</h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 21, 2023</p>
    </div>
  </div>
  
    <div>
    <div class="quarto-title-meta-heading">Modified</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">September 28, 2023</p>
    </div>
  </div>
    
  </div>
  

</header>

<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Disclaimer
</div>
</div>
<div class="callout-body-container callout-body">
<p>This is not a comprehensive list of Protein Modeling resources. Rather, it contains some of the references, methods, and online servers that I found useful through the years.</p>
<p>Most useful links (IMHO) are highlighted in bold. Other (many) tools may be useful for you, feel free to let me know and I’ll update the list.</p>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Links checked on 28/September/2023, let me know if you find any broken link.</p>
</div>
</div>
<section id="template-search" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> <strong>Template search</strong></h1>
<ul>
<li><p><strong>BLAST</strong>: <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="uri">https://blast.ncbi.nlm.nih.gov/Blast.cgi</a></p></li>
<li><p><strong>HHPRED</strong>: <a href="https://toolkit.tuebingen.mpg.de/tools/hhpred" class="uri">https://toolkit.tuebingen.mpg.de/tools/hhpred</a></p></li>
<li><p><strong>HHblits</strong>: <a href="https://toolkit.tuebingen.mpg.de/tools/hhblits" class="uri">https://toolkit.tuebingen.mpg.de/tools/hhblits</a></p></li>
<li><p><strong>JackHMMER</strong>: <a href="https://www.ebi.ac.uk/Tools/hmmer/search/jackhmmer" class="uri">https://www.ebi.ac.uk/Tools/hmmer/search/jackhmmer</a></p>
<ul>
<li>Ref. <a href="https://nar.oxfordjournals.org/content/46/W1/W200" class="uri">https://nar.oxfordjournals.org/content/46/W1/W200</a></li>
</ul></li>
</ul>
</section>
<section id="secondary-structure-and-1d-features-services" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> <strong>Secondary structure and 1D features services</strong></h1>
<ul>
<li><p><strong>Jpred4</strong>: <a href="http://www.compbio.dundee.ac.uk/jpred/" class="uri">http://www.compbio.dundee.ac.uk/jpred/</a></p>
<ul>
<li><p>Secondary structure prediction</p></li>
<li><p>Ref. <a href="http://nar.oxfordjournals.org/content/early/2015/04/16/nar.gkv332" class="uri">http://nar.oxfordjournals.org/content/early/2015/04/16/nar.gkv332</a></p></li>
</ul></li>
<li><p><strong>PsiPred</strong> : <a href="http://bioinf.cs.ucl.ac.uk/psipred/" class="uri">http://bioinf.cs.ucl.ac.uk/psipred/</a></p>
<ul>
<li><p>Secondary structure prediction</p></li>
<li><p>MEMSAT-SVM (Membrane Helix Prediction)</p></li>
<li><p>Ref: <a href="https://academic.oup.com/nar/article/47/W1/W402/5480136" class="uri">https://academic.oup.com/nar/article/47/W1/W402/5480136</a></p></li>
</ul></li>
<li><p><strong>MULTICOM Toolbox</strong>: <a href="http://sysbio.rnet.missouri.edu/multicom_toolbox/web%20services.html" class="uri">http://sysbio.rnet.missouri.edu/multicom_toolbox/web%20services.html</a></p>
<ul>
<li><p>Pulls together protein structure and structural feature prediction tools.</p></li>
<li><p>⚠️Server currently offline. Application available on GitHub: <a href="https://github.com/multicom-toolbox" class="uri">https://github.com/multicom-toolbox</a></p></li>
</ul></li>
<li><p><strong>SCRATCH</strong>: <a href="http://scratch.proteomics.ics.uci.edu/" class="uri">http://scratch.proteomics.ics.uci.edu/</a></p>
<ul>
<li>Metaserver with a wide range of 1D-3D prediction tools.</li>
</ul></li>
<li><p><strong>SYMPRED</strong>: <a href="http://www.ibi.vu.nl/programs/sympredwww/" class="uri">http://www.ibi.vu.nl/programs/sympredwww/</a></p>
<ul>
<li>Consensus secondary structure prediction metaserver</li>
</ul></li>
<li><p><strong>MESSA</strong>: <a href="http://prodata.swmed.edu/MESSA/MESSA.cgi" class="uri">http://prodata.swmed.edu/MESSA/MESSA.cgi</a></p>
<ul>
<li><p>1D features Metaserver</p></li>
<li><p>Ref.<a href="https://bmcbiol.biomedcentral.com/articles/10.1186/1741-7007-10-82" class="uri">https://bmcbiol.biomedcentral.com/articles/10.1186/1741-7007-10-82</a></p></li>
</ul></li>
<li><p><strong>DeepTMHMM</strong>: <a href="https://dtu.biolib.com/DeepTMHMM" class="uri">https://dtu.biolib.com/DeepTMHMM</a></p>
<ul>
<li><p>Deep-learning 2022 version of the classic TMHMM predictor of transmembrane helices</p></li>
<li><p>Ref. <a href="https://www.biorxiv.org/content/10.1101/2022.04.08.487609v1" class="uri">https://www.biorxiv.org/content/10.1101/2022.04.08.487609v1</a></p></li>
</ul></li>
<li><p><strong>Phobius</strong>: <a href="https://www.ebi.ac.uk/Tools/pfa/phobius/" class="uri">https://www.ebi.ac.uk/Tools/pfa/phobius/</a></p>
<ul>
<li>Prediction of transmembrane topology and signal peptides from the amino acid sequence of a protein.</li>
</ul></li>
<li><p><strong>TopCons</strong>: <a href="http://topcons.cbr.su.se/" class="uri">http://topcons.cbr.su.se/</a></p>
<ul>
<li><p>Consensus prediction of membrane protein topology and signal peptides</p></li>
<li><p>Ref. <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkv485" class="uri">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkv485</a></p></li>
</ul></li>
<li><p><strong>PSORT</strong>: <a href="https://psort.hgc.jp/" class="uri">https://psort.hgc.jp/</a></p>
<ul>
<li>Several localization signals</li>
</ul></li>
<li><p><strong>PRED-TMMB:</strong> <a href="http://bioinformatics.biol.uoa.gr/PRED-TMBB/" class="uri">http://bioinformatics.biol.uoa.gr/PRED-TMBB/</a></p>
<ul>
<li>Predicion of beta-barrel TMEs &amp; topology (HMM-based)</li>
</ul></li>
<li><p><strong>PROF-TMF</strong>: <a href="https://open.predictprotein.org/" class="uri">https://open.predictprotein.org/</a></p>
<ul>
<li>Predicion of beta-barrel TMEs (PredictProtein Suite)</li>
</ul></li>
<li><p><strong>NLStradamus</strong>: <a href="http://www.moseslab.csb.utoronto.ca/NLStradamus/" class="uri">http://www.moseslab.csb.utoronto.ca/NLStradamus/</a></p>
<ul>
<li>Prediction of NLS</li>
</ul></li>
<li><p><strong>Sulfinator:</strong> <a href="https://web.expasy.org/sulfinator/" class="uri">https://web.expasy.org/sulfinator/</a></p>
<ul>
<li>Prediction of disulfide bonds</li>
</ul></li>
<li><p><strong>DTU Tech resources</strong>: <a href="https://services.healthtech.dtu.dk/" class="uri">https://services.healthtech.dtu.dk/</a></p>
<ul>
<li>A collection of molecular biology tools, including a number of services for 1D features prediction, such as DeepTMHMM <span class="citation" data-cites="hallgren">(<a href="#ref-hallgren" role="doc-biblioref">Hallgren et al. 2022</a>)</span> or NetSurfP <span class="citation" data-cites="høie2022">(<a href="#ref-høie2022" role="doc-biblioref">Høie et al. 2022</a>)</span>.<br>
</li>
</ul></li>
</ul>
</section>
<section id="protein-modeling" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> <strong>Protein modeling</strong></h1>
<ul>
<li><p><strong>SwissModel</strong> <span class="citation" data-cites="waterhouse2018">(<a href="#ref-waterhouse2018" role="doc-biblioref">Waterhouse et al. 2018</a>)</span>: <a href="https://swissmodel.expasy.org/" class="uri">https://swissmodel.expasy.org/</a></p></li>
<li><p>PsiPRED <span class="citation" data-cites="buchan2019">(<a href="#ref-buchan2019" role="doc-biblioref">Buchan and Jones 2019</a>)</span>: <a href="http://bioinf.cs.ucl.ac.uk/psipred/" class="uri">http://bioinf.cs.ucl.ac.uk/psipred/</a></p>
<ul>
<li>Fold recognition server (among other tools)</li>
</ul></li>
<li><p><strong>RaptorX</strong> <span class="citation" data-cites="wang2017">(<a href="#ref-wang2017" role="doc-biblioref">Wang et al. 2017</a>)</span>: <a href="http://raptorx.uchicago.edu/ContactMap/" class="uri">http://raptorx.uchicago.edu/ContactMap/</a></p>
<ul>
<li>Distance-based Protein Folding Powered by Deep Learning</li>
</ul></li>
<li><p><strong>Phyre2</strong> <span class="citation" data-cites="kelley2015">(<a href="#ref-kelley2015" role="doc-biblioref">Kelley et al. 2015</a>)</span>: <a href="http://www.sbg.bio.ic.ac.uk/~phyre2/html/page.cgi?id=index" class="uri">http://www.sbg.bio.ic.ac.uk/~phyre2/html/page.cgi?id=index</a></p>
<ul>
<li>Fold recognition &amp; fragment assembler</li>
</ul></li>
<li><p><strong>Robetta</strong> server: <a href="https://robetta.bakerlab.org/" class="uri">https://robetta.bakerlab.org/</a></p>
<ul>
<li>Several methods, including RosettaCM <span class="citation" data-cites="song2013">(<a href="#ref-song2013" role="doc-biblioref">Song et al. 2013</a>)</span> and RoseTTAFOLD <span class="citation" data-cites="baek2021a baeka">(<a href="#ref-baek2021a" role="doc-biblioref">Baek et al. 2021</a>; <a href="#ref-baeka" role="doc-biblioref">Baek et al., n.d.</a>)</span></li>
</ul></li>
<li><p><strong>ModWeb</strong> <span class="citation" data-cites="pieper2014">(<a href="#ref-pieper2014" role="doc-biblioref">Pieper et al. 2014</a>)</span><strong>:</strong> <a href="https://modbase.compbio.ucsf.edu/modweb/" class="uri">https://modbase.compbio.ucsf.edu/modweb/</a></p>
<ul>
<li>Webserver using MODELLER (free registration required)</li>
</ul></li>
<li><p><strong>ColabFold</strong> <span class="citation" data-cites="mirdita2022">(<a href="#ref-mirdita2022" role="doc-biblioref">Mirdita et al. 2022</a>)</span>: <a href="https://github.com/sokrypton/ColabFold" class="uri">https://github.com/sokrypton/ColabFold</a></p>
<ul>
<li>Several notebooks available</li>
</ul></li>
<li><p><strong>DeepMind’s AlphaFold colab</strong>: <a href="https://colab.research.google.com/github/deepmind/alphafold/blob/main/notebooks/AlphaFold.ipynb" class="uri">https://colab.research.google.com/github/deepmind/alphafold/blob/main/notebooks/AlphaFold.ipynb</a></p></li>
<li><p><strong>Openfold</strong> <span class="citation" data-cites="ahdritz">(<a href="#ref-ahdritz" role="doc-biblioref">Ahdritz et al., n.d.</a>)</span> Colab: <a href="https://colab.research.google.com/github/aqlaboratory/openfold/blob/main/notebooks/OpenFold.ipyn" class="uri">https://colab.research.google.com/github/aqlaboratory/openfold/blob/main/notebooks/OpenFold.ipyn</a></p></li>
<li><p><strong>Unifold</strong> <span class="citation" data-cites="li">(<a href="#ref-li" role="doc-biblioref">Ziyao Li et al., n.d.</a>)</span> Colab: <a href="https://colab.research.google.com/github/dptech-corp/Uni-Fold/blob/main/notebooks/unifold.ipynb" class="uri">https://colab.research.google.com/github/dptech-corp/Uni-Fold/blob/main/notebooks/unifold.ipynb</a></p></li>
<li><p><strong>ESM Metagenomic Atlas</strong> (DDBB &amp; tools, <span class="citation" data-cites="lin2023">Lin et al. (<a href="#ref-lin2023" role="doc-biblioref">2023</a>)</span>): <a href="https://esmatlas.com/about" class="uri">https://esmatlas.com/about</a></p></li>
<li><p><strong>PymolFold</strong>: A handy Pymol script that calls the ESM-Fold API</p>
<ul>
<li><p><a href="https://github.com/JinyuanSun/PymolFold" class="uri">https://github.com/JinyuanSun/PymolFold</a></p></li>
<li><p>Web version: <a href="http://103.79.77.89:8501/" class="uri">http://103.79.77.89:8501/</a></p></li>
</ul></li>
<li><p><strong>Zhang server</strong>: <a href="https://zhanggroup.org/services/" class="uri">https://zhanggroup.org/services/</a></p>
<ul>
<li>Multiple methods, including I-TASSER <span class="citation" data-cites="roy2010">(<a href="#ref-roy2010" role="doc-biblioref">Roy, Kucukural, and Zhang 2010</a>)</span>, C-I-TASSER <span class="citation" data-cites="zheng2021">(<a href="#ref-zheng2021" role="doc-biblioref">Zheng et al. 2021</a>)</span> or I-TASSER-MTD, among others.</li>
</ul></li>
</ul>
</section>
<section id="model-quality-assessment" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> <strong>Model quality assessment</strong></h1>
<ul>
<li><p><strong>QMEAN &amp; QMEANDisCo</strong> <span class="citation" data-cites="studer2020">(<a href="#ref-studer2020" role="doc-biblioref">Studer et al. 2020</a>)</span>: <a href="https://swissmodel.expasy.org/qmean/" class="uri">https://swissmodel.expasy.org/qmean/</a></p></li>
<li><p><strong>Modfold</strong> <span class="citation" data-cites="mcguffin2021">(<a href="#ref-mcguffin2021" role="doc-biblioref">McGuffin et al. 2021</a>)</span><strong>:</strong> <a href="https://www.reading.ac.uk/bioinf/ModFOLD/" class="uri">https://www.reading.ac.uk/bioinf/ModFOLD/</a></p></li>
<li><p><strong>VoroMQA</strong> <span class="citation" data-cites="olechnovic2019">(<a href="#ref-olechnovic2019" role="doc-biblioref">Olechnovič and Venclovas 2019</a>)</span><strong>: <a href="http://bioinformatics.ibt.lt/wtsam/voromqa" class="uri">http://bioinformatics.ibt.lt/wtsam/voromqa</a></strong></p></li>
<li><p><strong>DeepUMQA</strong> <span class="citation" data-cites="guo2022">(<a href="#ref-guo2022" role="doc-biblioref">Guo et al. 2022</a>)</span>: <a href="http://zhanglab-bioinf.com/DeepUMQA/" class="uri">http://zhanglab-bioinf.com/DeepUMQA/</a></p></li>
</ul>
</section>
<section id="other-resources" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> <strong>Other resources</strong></h1>
<ul>
<li><p><strong>DALI</strong> <span class="citation" data-cites="holm2022">(<a href="#ref-holm2022" role="doc-biblioref">Holm 2022</a>)</span>: <a href="http://ekhidna2.biocenter.helsinki.fi/dali/" class="uri">http://ekhidna2.biocenter.helsinki.fi/dali/</a></p></li>
<li><p><strong>FoldSeek</strong> <span class="citation" data-cites="vankempen2023">(<a href="#ref-vankempen2023" role="doc-biblioref">Kempen et al. 2023</a>)</span>: <a href="https://search.foldseek.com/search" class="uri">https://search.foldseek.com/search</a></p>
<ul>
<li>Fast search for structural similarities</li>
</ul></li>
<li><p>FatCat <span class="citation" data-cites="li2020">(<a href="#ref-li2020" role="doc-biblioref">Zhanwen Li et al. 2020</a>)</span><strong>:</strong> <a href="http://fatcat.godziklab.org/fatcat/fatcat_pair.html" class="uri">http://fatcat.godziklab.org/fatcat/fatcat_pair.html</a></p>
<ul>
<li>Structural alignment tool.</li>
</ul></li>
<li><p>LGA: <a href="http://proteinmodel.org/AS2TS/LGA/lga.html" class="uri">http://proteinmodel.org/AS2TS/LGA/lga.html</a></p>
<ul>
<li><p>LGA&nbsp;program is being developed for&nbsp;structure&nbsp;comparative analysis of two selected 3D&nbsp;protein&nbsp;structures or fragments of 3D&nbsp;protein&nbsp;structures</p></li>
<li><p>Ref. <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkg571" class="uri">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkg571</a></p></li>
</ul></li>
<li><p>Galaxy Refine: <a href="http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=REFINE" class="uri">http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=REFINE</a></p>
<ul>
<li><p>A server for automatically refining protein model structures.</p></li>
<li><p>Ref. <a href="http://www.ncbi.nlm.nih.gov/pubmed/23737448" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/23737448</a></p></li>
</ul></li>
<li><p>Galaxy Refine2: <a href="http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=REFINE2"><u>galaxy.seoklab.org/cgi-bin/submit.cgi?type=REFINE2</u></a></p>
<ul>
<li><p>&lt;300 aa</p></li>
<li><p>Ref:<a href="https://academic.oup.com/nar/article/47/W1/W451/5475172" class="uri">https://academic.oup.com/nar/article/47/W1/W451/5475172</a></p></li>
</ul></li>
<li><p>HOMCOS: <a href="http://homcos.pdbj.org/" class="uri">http://homcos.pdbj.org/</a></p>
<ul>
<li><p>Modeling of 3D Structures of Complexes</p></li>
<li><p>Ref. <a href="http://link.springer.com/article/10.1007/s10969-016-9208-y?wt_mc=Internal.Event.1.SEM.ArticleAuthorOnlineFirst" class="uri">http://link.springer.com/article/10.1007/s10969-016-9208-y?wt_mc=Internal.Event.1.SEM.ArticleAuthorOnlineFirst</a></p></li>
</ul></li>
<li><p>pyDockWEB: <a href="https://life.bsc.es/pid/pydockweb" class="uri">https://life.bsc.es/pid/pydockweb</a></p>
<ul>
<li>Prediction of protein-protein interactions (hetero-oligomers modeling)</li>
</ul></li>
<li><p>Hex: <a href="http://hexserver.loria.fr/" class="uri">http://hexserver.loria.fr/</a></p>
<ul>
<li><p>Prediction of protein-protein interactions (homo- &amp; hetero-oligomers modeling)</p></li>
<li><p>Ref. <a href="https://onlinelibrary.wiley.com/doi/full/10.1002/prot.24433" class="uri">https://onlinelibrary.wiley.com/doi/full/10.1002/prot.24433</a></p></li>
</ul></li>
<li><p>DaReUs-Loop<strong>:</strong> <a href="https://bioserv.rpbs.univ-paris-diderot.fr/services/DaReUS-Loop/" class="uri">https://bioserv.rpbs.univ-paris-diderot.fr/services/DaReUS-Loop/</a></p>
<ul>
<li><p>Modeling loops</p></li>
<li><p>Ref. <a href="https://www.nature.com/articles/s41598-018-32079-w" class="uri">https://www.nature.com/articles/s41598-018-32079-w</a></p></li>
</ul></li>
<li><p>GalaxyLoop: <a href="http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=LOOP" class="uri">http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=LOOP</a></p>
<ul>
<li><p>Loops refining</p></li>
<li><p>Ref. <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0113811" class="uri">https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0113811</a></p></li>
</ul></li>
<li><p>Expasy proteomics tools: <a href="https://www.expasy.org/proteomics" class="uri">https://www.expasy.org/proteomics</a></p>
<ul>
<li>List of qMultiple links and resources</li>
</ul></li>
<li><p>Protein modeling links @Sali Lab: <a href="https://salilab.org/bioinformatics_resources.html" class="uri">https://salilab.org/bioinformatics_resources.html</a></p></li>
<li><p>Homology modeling @Proteopedia:</p>
<ul>
<li><p><a href="https://proteopedia.org/wiki/index.php/Homology_model" class="uri">https://proteopedia.org/wiki/index.php/Homology_model</a></p></li>
<li><p><a href="https://proteopedia.org/wiki/index.php/User:Wayne_Decatur/Homology_Modeling" class="uri">https://proteopedia.org/wiki/index.php/User:Wayne_Decatur/Homology_Modeling</a></p></li>
</ul></li>
<li><p>List of protein structure prediction software @Wikipedia: <a href="https://en.wikipedia.org/wiki/List_of_protein_structure_prediction_software" class="uri">https://en.wikipedia.org/wiki/List_of_protein_structure_prediction_software</a></p></li>
</ul>



</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-ahdritz" class="csl-entry" role="listitem">
Ahdritz, Gustaf, Nazim Bouatta, Christina Floristean, Sachin Kadyan, Qinghui Xia, William Gerecke, Timothy J. O’Donnell, et al. n.d. <span>“OpenFold: Retraining AlphaFold2 Yields New Insights into Its Learning Mechanisms and Capacity for Generalization.”</span> <a href="https://doi.org/10.1101/2022.11.20.517210">https://doi.org/10.1101/2022.11.20.517210</a>.
</div>
<div id="ref-baeka" class="csl-entry" role="listitem">
Baek, Minkyung, Ivan Anishchenko, Ian R. Humphreys, Qian Cong, David Baker, and Frank DiMaio. n.d. <span>“Efficient and Accurate Prediction of Protein Structure Using RoseTTAFold2.”</span> <a href="https://doi.org/10.1101/2023.05.24.542179">https://doi.org/10.1101/2023.05.24.542179</a>.
</div>
<div id="ref-baek2021a" class="csl-entry" role="listitem">
Baek, Minkyung, Frank DiMaio, Ivan Anishchenko, Justas Dauparas, Sergey Ovchinnikov, Gyu Rie Lee, Jue Wang, et al. 2021. <span>“Accurate prediction of protein structures and interactions using a three-track neural network.”</span> <em>Science (New York, N.Y.)</em> 373 (6557): 871–76. <a href="https://doi.org/10.1126/science.abj8754">https://doi.org/10.1126/science.abj8754</a>.
</div>
<div id="ref-buchan2019" class="csl-entry" role="listitem">
Buchan, Daniel W A, and David T Jones. 2019. <span>“The PSIPRED Protein Analysis Workbench: 20 Years On.”</span> <em>Nucleic Acids Research</em> 47 (W1): W402–7. <a href="https://doi.org/10.1093/nar/gkz297">https://doi.org/10.1093/nar/gkz297</a>.
</div>
<div id="ref-guo2022" class="csl-entry" role="listitem">
Guo, Sai-Sai, Jun Liu, Xiao-Gen Zhou, and Gui-Jun Zhang. 2022. <span>“DeepUMQA: ultrafast shape recognition-based protein model quality assessment using deep learning.”</span> <em>Bioinformatics (Oxford, England)</em> 38 (7): 1895–1903. <a href="https://doi.org/10.1093/bioinformatics/btac056">https://doi.org/10.1093/bioinformatics/btac056</a>.
</div>
<div id="ref-hallgren" class="csl-entry" role="listitem">
Hallgren, Jeppe, Konstantinos D. Tsirigos, Mads Damgaard Pedersen, José Juan Almagro Armenteros, Paolo Marcatili, Henrik Nielsen, Anders Krogh, and Ole Winther. 2022. <span>“DeepTMHMM Predicts Alpha and Beta Transmembrane Proteins Using Deep Neural Networks.”</span> <a href="https://doi.org/10.1101/2022.04.08.487609">https://doi.org/10.1101/2022.04.08.487609</a>.
</div>
<div id="ref-høie2022" class="csl-entry" role="listitem">
Høie, Magnus Haraldson, Erik Nicolas Kiehl, Bent Petersen, Morten Nielsen, Ole Winther, Henrik Nielsen, Jeppe Hallgren, and Paolo Marcatili. 2022. <span>“NetSurfP-3.0: Accurate and Fast Prediction of Protein Structural Features by Protein Language Models and Deep Learning.”</span> <em>Nucleic Acids Research</em> 50 (W1): W510–15. <a href="https://doi.org/10.1093/nar/gkac439">https://doi.org/10.1093/nar/gkac439</a>.
</div>
<div id="ref-holm2022" class="csl-entry" role="listitem">
Holm, Liisa. 2022. <span>“Dali Server: Structural Unification of Protein Families.”</span> <em>Nucleic Acids Research</em> 50 (W1): W210–15. <a href="https://doi.org/10.1093/nar/gkac387">https://doi.org/10.1093/nar/gkac387</a>.
</div>
<div id="ref-kelley2015" class="csl-entry" role="listitem">
Kelley, L. A., S. Mezulis, C. M. Yates, M. N. Wass, and M. J. Sternberg. 2015. <span>“The Phyre2 Web Portal for Protein Modeling, Prediction and Analysis.”</span> <em>Nat Protoc</em> 10 (6): 845–58. <a href="https://doi.org/10.1038/nprot.2015.053">https://doi.org/10.1038/nprot.2015.053</a>.
</div>
<div id="ref-vankempen2023" class="csl-entry" role="listitem">
Kempen, Michel van, Stephanie S. Kim, Charlotte Tumescheit, Milot Mirdita, Jeongjae Lee, Cameron L. M. Gilchrist, Johannes Söding, and Martin Steinegger. 2023. <span>“Fast and Accurate Protein Structure Search with Foldseek.”</span> <em>Nature Biotechnology</em>, May, 1–4. <a href="https://doi.org/10.1038/s41587-023-01773-0">https://doi.org/10.1038/s41587-023-01773-0</a>.
</div>
<div id="ref-li2020" class="csl-entry" role="listitem">
Li, Zhanwen, Lukasz Jaroszewski, Mallika Iyer, Mayya Sedova, and Adam Godzik. 2020. <span>“FATCAT 2.0: towards a better understanding of the structural diversity of proteins.”</span> <em>Nucleic Acids Research</em> 48 (W1): W60–64. <a href="https://doi.org/10.1093/nar/gkaa443">https://doi.org/10.1093/nar/gkaa443</a>.
</div>
<div id="ref-li" class="csl-entry" role="listitem">
Li, Ziyao, Xuyang Liu, Weijie Chen, Fan Shen, Hangrui Bi, Guolin Ke, and Linfeng Zhang. n.d. <span>“Uni-Fold: An Open-Source Platform for Developing Protein Folding Models Beyond AlphaFold.”</span> <a href="https://doi.org/10.1101/2022.08.04.502811">https://doi.org/10.1101/2022.08.04.502811</a>.
</div>
<div id="ref-lin2023" class="csl-entry" role="listitem">
Lin, Zeming, Halil Akin, Roshan Rao, Brian Hie, Zhongkai Zhu, Wenting Lu, Nikita Smetanin, et al. 2023. <span>“Evolutionary-Scale Prediction of Atomic-Level Protein Structure with a Language Model.”</span> <em>Science</em> 379 (6637): 1123–30. <a href="https://doi.org/10.1126/science.ade2574">https://doi.org/10.1126/science.ade2574</a>.
</div>
<div id="ref-mcguffin2021" class="csl-entry" role="listitem">
McGuffin, Liam J., Fahd M. F. Aldowsari, Shuaa M. A. Alharbi, and Recep Adiyaman. 2021. <span>“ModFOLD8: accurate global and local quality estimates for 3D protein models.”</span> <em>Nucleic Acids Research</em> 49 (W1): W425–30. <a href="https://doi.org/10.1093/nar/gkab321">https://doi.org/10.1093/nar/gkab321</a>.
</div>
<div id="ref-mirdita2022" class="csl-entry" role="listitem">
Mirdita, Milot, Konstantin Schütze, Yoshitaka Moriwaki, Lim Heo, Sergey Ovchinnikov, and Martin Steinegger. 2022. <span>“ColabFold: making protein folding accessible to all.”</span> <em>Nature Methods</em> 19 (6): 679–82. <a href="https://doi.org/10.1038/s41592-022-01488-1">https://doi.org/10.1038/s41592-022-01488-1</a>.
</div>
<div id="ref-olechnovic2019" class="csl-entry" role="listitem">
Olechnovič, Kliment, and Česlovas Venclovas. 2019. <span>“VoroMQA web server for assessing three-dimensional structures of proteins and protein complexes.”</span> <em>Nucleic Acids Research</em> 47 (W1): W437–42. <a href="https://doi.org/10.1093/nar/gkz367">https://doi.org/10.1093/nar/gkz367</a>.
</div>
<div id="ref-pieper2014" class="csl-entry" role="listitem">
Pieper, Ursula, Benjamin M. Webb, Guang Qiang Dong, Dina Schneidman-Duhovny, Hao Fan, Seung Joong Kim, Natalia Khuri, et al. 2014. <span>“ModBase, a database of annotated comparative protein structure models and associated resources.”</span> <em>Nucleic Acids Research</em> 42 (Database issue): D336–346. <a href="https://doi.org/10.1093/nar/gkt1144">https://doi.org/10.1093/nar/gkt1144</a>.
</div>
<div id="ref-roy2010" class="csl-entry" role="listitem">
Roy, A., A. Kucukural, and Y. Zhang. 2010. <span>“I-TASSER: a unified platform for automated protein structure and function prediction.”</span> <em>Nat Protoc</em> 5 (4): 725–38. <a href="https://doi.org/nprot.2010.5 [pii] 10.1038/nprot.2010.5">https://doi.org/nprot.2010.5 [pii] 10.1038/nprot.2010.5</a>.
</div>
<div id="ref-song2013" class="csl-entry" role="listitem">
Song, Yifan, Frank DiMaio, Ray Yu-Ruei Wang, David Kim, Chris Miles, TJ Brunette, James Thompson, and David Baker. 2013. <span>“High-Resolution Comparative Modeling with RosettaCM.”</span> <em>Structure</em> 21 (10): 1735–42. https://doi.org/<a href="https://doi.org/10.1016/j.str.2013.08.005">https://doi.org/10.1016/j.str.2013.08.005</a>.
</div>
<div id="ref-studer2020" class="csl-entry" role="listitem">
Studer, Gabriel, Christine Rempfer, Andrew M. Waterhouse, Rafal Gumienny, Juergen Haas, and Torsten Schwede. 2020. <span>“QMEANDisCo-distance constraints applied on model quality estimation.”</span> <em>Bioinformatics (Oxford, England)</em> 36 (6): 1765–71. <a href="https://doi.org/10.1093/bioinformatics/btz828">https://doi.org/10.1093/bioinformatics/btz828</a>.
</div>
<div id="ref-wang2017" class="csl-entry" role="listitem">
Wang, Sheng, Siqi Sun, Zhen Li, Renyu Zhang, and Jinbo Xu. 2017. <span>“Accurate De Novo Prediction of Protein Contact Map by Ultra-Deep Learning Model.”</span> <em>PLoS computational biology</em> 13 (1): e1005324. <a href="https://doi.org/10.1371/journal.pcbi.1005324">https://doi.org/10.1371/journal.pcbi.1005324</a>.
</div>
<div id="ref-waterhouse2018" class="csl-entry" role="listitem">
Waterhouse, Andrew, Martino Bertoni, Stefan Bienert, Gabriel Studer, Gerardo Tauriello, Rafal Gumienny, Florian T Heer, et al. 2018. <span>“SWISS-MODEL: Homology Modelling of Protein Structures and Complexes.”</span> <em>Nucleic Acids Research</em> 46 (W1): W296–303. <a href="https://doi.org/10.1093/nar/gky427">https://doi.org/10.1093/nar/gky427</a>.
</div>
<div id="ref-zheng2021" class="csl-entry" role="listitem">
Zheng, Wei, Chengxin Zhang, Yang Li, Robin Pearce, Eric W. Bell, and Yang Zhang. 2021. <span>“Folding non-homologous proteins by coupling deep-learning contact maps with I-TASSER assembly simulations.”</span> <em>Cell Reports Methods</em> 1 (3): 100014. <a href="https://doi.org/10.1016/j.crmeth.2021.100014">https://doi.org/10.1016/j.crmeth.2021.100014</a>.
</div>
</div></section></div></main> <!-- /main -->
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
<nav class="page-navigation column-body">
  <div class="nav-page nav-page-previous">
      <a href="./ai.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Protein modeling in the AlphaFold era</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">Modesto Redrejo Rodríguez, 2023</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://creativecommons.org/licenses/by-nc/2.0/">
      <i class="bi bi-badge-cc-fill" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://twitter.com/mredrejo"><iconify-icon inline="" icon="fa6-brands:x-twitter"></iconify-icon></a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link active" href="https://github.com/StrBio/strbio.github.io" aria-current="page">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



<script src="site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>