<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-08-21">

<title>Structural Bioinformatics 2023-24 - Protein Structures</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./homology.html" rel="next">
<link href="./ddbb.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/iconify-1.0.0-beta.2/iconify-icon.min.js"></script>
<script src="site_libs/quarto-contrib/molstar-v3.13.0/molstar.js"></script>
<link href="site_libs/quarto-contrib/molstar-v3.13.0/molstar.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/molstar-v3.13.0/app-container.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>


<link rel="stylesheet" href="styles.css">
<link rel="stylesheet" href="include/webex.css">
<meta name="twitter:title" content="Structural Bioinformatics 2023-24 - Protein Structures">
<meta name="twitter:description" content="">
<meta name="twitter:image" content="pics/magritte.png">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./intro.html">Background</a></li><li class="breadcrumb-item"><a href="./pdb.html">Protein Structures</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Structural Bioinformatics 2023-24</a> 
        <div class="sidebar-tools-main">
    <a href="mailto:modesto.redrejo@uam.es" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-envelope-paper"></i></a>
    <a href="https://github.com/StrBio/strbio.github.io" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">StrBio 2023-24</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Background</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ddbb.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pdb.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Protein Structures</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Modeling</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./homology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Homology Modeling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1d.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1D Features</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./advanced.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Advanced homology modeling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ai.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein modeling in the AlphaFold era</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./links.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Modeling Resources</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul class="collapse">
  <li><a href="#obtaining-and-working-with-protein-structures" id="toc-obtaining-and-working-with-protein-structures" class="nav-link active" data-scroll-target="#obtaining-and-working-with-protein-structures"><span class="header-section-number">1</span> Obtaining and working with protein structures</a></li>
  <li><a href="#experimental-determination-of-protein-structures" id="toc-experimental-determination-of-protein-structures" class="nav-link" data-scroll-target="#experimental-determination-of-protein-structures"><span class="header-section-number">2</span> Experimental determination of protein structures</a>
  <ul class="collapse">
  <li><a href="#x-ray-crystallography-or-single-crystal-x-ray-diffraction" id="toc-x-ray-crystallography-or-single-crystal-x-ray-diffraction" class="nav-link" data-scroll-target="#x-ray-crystallography-or-single-crystal-x-ray-diffraction"><span class="header-section-number">2.1</span> X-Ray crystallography or single crystal X-ray diffraction</a></li>
  <li><a href="#nuclear-magnetic-resonance" id="toc-nuclear-magnetic-resonance" class="nav-link" data-scroll-target="#nuclear-magnetic-resonance"><span class="header-section-number">2.2</span> Nuclear Magnetic Resonance</a></li>
  <li><a href="#electron-cryomicroscopy" id="toc-electron-cryomicroscopy" class="nav-link" data-scroll-target="#electron-cryomicroscopy"><span class="header-section-number">2.3</span> Electron cryomicroscopy</a></li>
  </ul></li>
  <li><a href="#sec-assess" id="toc-sec-assess" class="nav-link" data-scroll-target="#sec-assess"><span class="header-section-number">3</span> Structural quality assurance</a>
  <ul class="collapse">
  <li><a href="#global-parameters-in-experimentally-based-structures" id="toc-global-parameters-in-experimentally-based-structures" class="nav-link" data-scroll-target="#global-parameters-in-experimentally-based-structures"><span class="header-section-number">3.1</span> Global parameters in experimentally-based structures</a></li>
  <li><a href="#stereochemical-parameters" id="toc-stereochemical-parameters" class="nav-link" data-scroll-target="#stereochemical-parameters"><span class="header-section-number">3.2</span> Stereochemical parameters</a></li>
  </ul></li>
  <li><a href="#sec-apps" id="toc-sec-apps" class="nav-link" data-scroll-target="#sec-apps"><span class="header-section-number">4</span> Protein structure display</a>
  <ul class="collapse">
  <li><a href="#protein-structure-file-formats" id="toc-protein-structure-file-formats" class="nav-link" data-scroll-target="#protein-structure-file-formats"><span class="header-section-number">4.1</span> Protein structure file formats</a></li>
  <li><a href="#biological-macromolecules-display-applications" id="toc-biological-macromolecules-display-applications" class="nav-link" data-scroll-target="#biological-macromolecules-display-applications"><span class="header-section-number">4.2</span> Biological macromolecules display applications</a></li>
  </ul></li>
  <li><a href="#pymol-practice" id="toc-pymol-practice" class="nav-link" data-scroll-target="#pymol-practice"><span class="header-section-number">5</span> <span style="color:green"><strong>PyMOL Practice</strong></span></a>
  <ul class="collapse">
  <li><a href="#part-a-a-10-steps-self-guided-practice" id="toc-part-a-a-10-steps-self-guided-practice" class="nav-link" data-scroll-target="#part-a-a-10-steps-self-guided-practice"><span class="header-section-number">5.1</span> PART A: A 10-steps self-guided practice</a></li>
  <li><a href="#part-b-pymol-challenge" id="toc-part-b-pymol-challenge" class="nav-link" data-scroll-target="#part-b-pymol-challenge"><span class="header-section-number">5.2</span> PART B: PyMOL Challenge</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Protein Structures</h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 21, 2023</p>
    </div>
  </div>
  
    <div>
    <div class="quarto-title-meta-heading">Modified</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">August 29, 2023</p>
    </div>
  </div>
    
  </div>
  

</header>

<section id="obtaining-and-working-with-protein-structures" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Obtaining and working with protein structures</h1>
<div id="fig-magritte" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pics/magritte.png" title="Ceci n'est pas une proteine" class="figure img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;1: Ceci n’est pas une proteine. Source: <a href="https://swissmodel.expasy.org/static/course/files/PartIII_quality_assesment.pdf">SwissModel site</a>.</figcaption>
</figure>
</div>
<p>The surrealist Belgian painter René Magritte created a collection of surrealistic paintings entitled <a href="https://en.wikipedia.org/wiki/The_Treachery_of_Images" title="Magritte"><strong><em>La trahison des images</em></strong> (1928–1929)</a>. The most renowned of those paintings show a smoking pipe and the following caption underneath: <em>“Ceci n’est pas une pipe”</em> (This is not a pipe). Yes, indeed! It is actually the painting of a pipe.</p>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Warning for current and future structural biologists
</div>
</div>
<div class="callout-body-container callout-body">
<p>A picture of a protein, or a computer file containing the coordinates of a protein structure, is not a protein. It is a representation of ONE possible structure of that protein.</p>
</div>
</div>
<p>Even experimentally determined structures have two major limitations that we should always keep in mind: (1) they are a fixed structure (except RMN-based) whereas proteins <em>in vivo</em> are flexible and dynamic and (2) they are subject to experimental error and they often contain regions of low confidence (see below <a href="#sec-assess">Section&nbsp;3</a>). Moreover, even experimentally obtained macromolecular structures are to some extent models, with a variable ratio between experimental data and computational predictions to match the experimental data (X-ray diffraction, cryo-EM density maps, NMR, SAXS, FRET…) with previously known structures or models. Of course, this does not mean that protein structures are useless, they can be very useful, but we must be aware of both the limitations and the applications.</p>
</section>
<section id="experimental-determination-of-protein-structures" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Experimental determination of protein structures</h1>
<p>The structure helps to understand the molecular mechanism of protein function at a higher level of detail. The 3D representation can help orient different domains/motifs/residues of interest. This can be critical for understanding population or pathogenic variants, drug design, and protein engineering. Moreover, the structures can also help in predicting protein function and evolution because they are more conserved than sequence, i.e., protein structure space is smaller than sequence space. However, obtaining detailed and reliable structural data can be technically difficult and time consuming and as we will see, protein structure modeling can often be a good complement or alternative. Experimentally obtained structures usually rely on three alternative techniques, X-ray crystallography, nuclear magnetic resonance (RMN), or electron cryomicroscopy (CryoEM).</p>
<section id="x-ray-crystallography-or-single-crystal-x-ray-diffraction" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="x-ray-crystallography-or-single-crystal-x-ray-diffraction"><span class="header-section-number">2.1</span> X-Ray crystallography or single crystal X-ray diffraction</h2>
<p>X-Ray crystallography or single-crystal X-ray diffraction, is a method for determining the atomic structure of molecules in regular, crystalline structures. It requires the generation of a crystal of the molecule of interest, which is then mounted on a goniometer and illuminated or irradiated with a focused beam of X-rays. The diffraction pattern of the X-rays on the other side of the crystal allows determination of the positions of the atoms, as well as their chemical bonds, their crystallographic disorder, and various other information. However, the link between the diffraction pattern and the electron density is not trivial and requires some complex maths, called <a href="http://pruffle.mit.edu/atomiccontrol/education/xray/fourier.php">Fourier transforms</a>.</p>
<div id="fig-xray" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="images/paste-F430F5B1.png" class="figure img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;2: Schematic workflow of X-ray crystallography. From Creative Structure <a href="https://www.creative-biostructure.com/comparison-of-crystallography-nmr-and-em_6.htm">website</a>.</figcaption>
</figure>
</div>
<p>X-ray diffraction is a powerful method that allows obtaining high-resolution atomic-level structures of soluble or membrane proteins, either as apoenzymes or as holoenzymes bound to a substrate, cofactor or drugs. However, the sample must be crystallizable (i.e., homogeneous), which requires a considerable amount of very pure protein. Another disadvantage of X-ray structures is that, as mentioned earlier, you only get one (or very few) static forms of the protein and the location of hydrogen atoms cannot be determined by conventional diffraction methods —the fact that they have only one electron makes them very hard to detect with X-rays accurately because X-rays scatter at the electron density. They can be predicted, but that still hinders some chemical analysis.</p>
</section>
<section id="nuclear-magnetic-resonance" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="nuclear-magnetic-resonance"><span class="header-section-number">2.2</span> Nuclear Magnetic Resonance</h2>
<p>All atomic nuclei are charged, fast spinning particles, which gave rise to resonance frequencies that are different for each atom. Therefore, if we apply a magnetic field we can obtain an electromagnetic signal with a characteristic frequency of the magnetic field at the nucleus. This is the basis of nuclear magnetic resonance (NMR).</p>
<p>We should also remember that the motion of the nucleus is not isolated and it interacts with the surrounding atoms both intra- and intermolecularly. Nuclear magnetic resonance spectroscopy can thus provide structural information about a particular molecule. Taking a protein as an example, its secondary structures, such as α-helix, β-sheet or turn, reflect the different arrangement of the main chain atoms of protein molecules in three dimensions. The spacing of the atomic nuclei in the different secondary structures, the interaction between nuclei, and the dynamic characteristics of polypeptide segments all directly reflect the three-dimensional structure of proteins. These nuclear features all contribute to spectroscopic behaviors of the analyzed sample, thus providing characteristic NMR signals. Interpretation of these signals by computer-aided methods leads to deciphering of the three-dimensional structure.</p>
<div id="fig-rmn" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="images/paste-2013F0AC.png" class="figure img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;3: Basis of nuclear magnetic resonance. From Creative Structure <a href="https://www.creative-biostructure.com/comparison-of-crystallography-nmr-and-em_6.htm">website</a>.</figcaption>
</figure>
</div>
<p>The most important feature of the NMR method is that the three-dimensional structure of macromolecules in the natural state can be measured directly in solution, and NMR may provide unique information about dynamics and intermolecular interactions. The resolution of the macromolecular three-dimensional structure can be as low as sub nanometer. However, the NMR spectrum of biomolecules with large molecular weight is very complicated and difficult to interpret, thereby limiting the application of NMR in analyzing large biomolecules, often below 20-30 kDa (see <a href="#fig-experimental">Figure&nbsp;4</a>). Additionally, this technique requires relatively large amounts of pure samples (on the order of several mg) to achieve a reasonable signal to noise level.</p>
<div id="fig-experimental" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://doi.org/10.1016/j.jsb.2022.107841"><img src="images/paste-FA73AF1E.png" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">Figure&nbsp;4: Coverage of molecular weight by structural technique. From .</figcaption>
</figure>
</div>
</section>
<section id="electron-cryomicroscopy" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="electron-cryomicroscopy"><span class="header-section-number">2.3</span> Electron cryomicroscopy</h2>
<p>The essential mechanism of Cryo-EM is the same of any electron microscopy method, i. e., electron scattering. Samples are prepared through cryopreservation prior to analysis. The, a source electrons is used as a light source to measure the sample. After the electron beam passes through the sample, a complex lens system converts the scattered signal into a magnified image recorded on the detector. A key subsequent step is signal processing, that transform thousands of images of the particles in any orientation into a three-dimensional structure of the sample.</p>
<div id="fig-cryoEM" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="images/paste-5E29F580.png" class="figure img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;5: The process of Cryo-EM single particle analysis technique. From Creative Structure <a href="https://www.creative-biostructure.com/comparison-of-crystallography-nmr-and-em_6.htm">website</a>.</figcaption>
</figure>
</div>
<p>The use of electron microscopy methods for structural biology was traditionally limited to very large macromolecular complexes, like viral capsids, and only recently it could be used for smaller particles (see <a href="#fig-experimental">Figure&nbsp;4</a>). The number of protein structures being determined by cryo-electron microscopy is growing at an explosive rate in the last 5-10 years. This is thanks to several technical improvements in the technique, spanning sample preparation, analysis and processing that allow obtaining pictures at the atomic level <span class="citation" data-cites="callaway2020">(<a href="#ref-callaway2020" role="doc-biblioref">Callaway 2020</a>)</span>. This advances were acknowledged by the <a href="https://www.nobelprize.org/prizes/chemistry/2017/press-release/">2017 Nobel Prize in Chemistry</a> to Jacques Dubochet, Joachim Frank and Richard Henderson.</p>
<div id="fig-cryoEM2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="images/paste-E5E5AE75.png" class="figure img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;6: Cryo-electron microscopy revolution. From Creative Structure <a href="https://www.creative-biostructure.com/comparison-of-crystallography-nmr-and-em_6.htm">website</a>.</figcaption>
</figure>
</div>
<section id="tip" class="level3 callout-info" data-collapse="true">
<h3 class="anchored" data-anchor-id="tip">Tip</h3>
<p>Check the already classic article by <span class="citation" data-cites="egelman2016">Egelman (<a href="#ref-egelman2016" role="doc-biblioref">2016</a>)</span> for more a detailed info. And <a href="https://www.chemistryworld.com/news/explainer-what-is-cryo-electron-microscopy/3008091.article">here</a> for a great outreaching article after the Nobel Prize.</p>
</section>
<p>CryoEM is widely use nowadays because, particularly for large molecular complexes or viral particles. Structures can be generated quickly, as it does not require a high amount of protein and it can generate good data even in the presence of impurities. However, new generation microscopes are only affordable by large institutions and small particles can have a high level of noise. Moreover, processing a large amount of images can be limiting to obtain high-quality structures.</p>
</section>
</section>
<section id="sec-assess" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Structural quality assurance</h1>
<p>As pointed out at the beginning of this section, any structure, regardless of its origin or method of determination, is subject to error. Structures determined experimentally are actually models that were constructed to match with the experimental data. The quality of the original data and the precision with which the experiments were performed determine the reliability of the structural results. As in any other scientific discipline, experiments performed independently can lead to related models of the same molecule, but there are almost always differences; nevertheless, both can be good models.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Extra info
</div>
</div>
<div class="callout-body-container callout-body">
<p>Check the detailed documentation about PDB validation report <a href="https://www.wwpdb.org/validation/XrayValidationReportHelp">here</a>.</p>
</div>
</div>
<section id="global-parameters-in-experimentally-based-structures" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="global-parameters-in-experimentally-based-structures"><span class="header-section-number">3.1</span> Global parameters in experimentally-based structures</h2>
<p>There are a number of different parameters that help us understand the quality and reliability of a structure. First, the <strong>resolution</strong> is a good indicator of the level of detail of the structure, as it can greatly affect affect how the experimental data are modeled.</p>
<div id="fig-reso" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="images/paste-C3031EBE.png" class="figure img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;7: The effect of resolution on the quality of the electron density. The Tyr100 residue from concanavalin A as found in the indicated PDB structures at 3 Å, 2 Å and 1.2 Å. Reproduction of Figure 14.5 from <span class="citation" data-cites="structur">Gu and Bourne (<a href="#ref-structur" role="doc-biblioref">2009</a>)</span> rendered with Pymol (see <a href="../concanavalin.pse">concanavalin.pse</a> and <a href="../concanavalin.txt">concanavalin.txt</a> in the Repo for details about the picture display).</figcaption>
</figure>
</div>
<p>    </p><div id="app-mol-star_state.molj" class="molstar-app"></div>
    <script type="text/javascript">
    molstar.Viewer.create("app-mol-star_state.molj", {"viewportShowSelectionMode":false,"layoutShowLog":false,"layoutIsExpanded":false,"emdbProvider":"rcsb","layoutShowLeftPanel":true,"layoutShowControls":false,"viewportShowAnimation":true,"pdbProvider":"rcsb","layoutShowRemoteState":false,"layoutShowSequence":false,"viewportShowExpand":true}).then(viewer => {
    viewer.loadSnapshotFromUrl(url="mol-star_state.molj", "molj");    });
    </script>
     Embedded reproduction of the <a href="#fig-reso">Figure&nbsp;7</a> with <em>Mol*</em>, which allow you to explore the structures.<p></p>
<p>Another important parameter is the <strong><em>R</em>-factor</strong>, which is the difference between the structure factors calculated from the model and those obtained from the experimental data. That is, the <em>R</em>-factor is the deviation between the calculated diffraction pattern of the model and the original experimental diffraction pattern. Typically, good structures with a resolution of 1-3 Å, have an <em>R-</em>factor of 0.2 (i.e., 20% of deviation). However, it should be noted that this factor is usually reduced after iterative refinement, which downplays its use as an indicator of reliability. A more reliable factor is the <strong><em>R<sub>free</sub></em></strong> factor. This is less susceptible to manipulation during refinement, as it is based on only a small portion of the experimental data (5-10%) that is not used during the refinement phase.</p>
<p>A more intuitive, but only qualitative, way to understand the precision of the coordinates of a given atom is the <em>B-</em>factor. The temperature value or <em>B-</em>factor correlates with the position errors, although its mathematical definition is more complex. Normal values for a B-factor are in the range of 14-30, while values above 30 usually indicate that the atom is in a flexible or disordered region, and atoms with a <em>B-</em>factor above 40 are often ruled out as too unreliable.</p>
<p>The root-mean-squared deviation (<strong>RMSD,</strong> <a href="ddbb.html#sec-alignment">see Structure alignment section</a>) is a traditional estimator of the quality of NMR-solved structures. Regions with high RMSD values are those that are less defined by data. However, it should also be noted that this parameter can be also misleading, as it is highly dependent on the procedure used to generate and select the data that is submitted to the PDB. An experimentalist could reduce the RMSD by selecting the “best” few structures for deposition from a much larger draft. Note that the RMSD has many other applications, like comparing different structures or models from the same or related sequences.</p>
<p>In recent years, with the increase of quantity and quality of EM structures, new parameters have also been proposed. One of them, the <strong><em>Q-</em>factor</strong> was recently introduced for <a href="https://www.rcsb.org/news/feature/62de9e5235ec5bb4ddb19a43">validation of 3DEM/PDB structures</a>. Briefly, the Q-factor score calculates the resolvability of atoms by measuring the similarity of the map values around each atom relative to a Gaussian-like function for a well resolved atom. A Q score of 1 means that the similarity is perfect, while a value close to 0 indicates low similarity. If the atom is not well placed in the map, a negative Q value can be given. Therefore, Q-factor values in the reports range from -1 to +1.</p>
</section>
<section id="stereochemical-parameters" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="stereochemical-parameters"><span class="header-section-number">3.2</span> Stereochemical parameters</h2>
<p>Since all structural models contain some degree of error and some of the global modeling parameters may be controversial, we can analyze the geometry, stereochemistry, and other structural properties of the model to evaluate structural models. These parameters compare a given structure to what is already known about that type of molecule based on our knowledge from high-resolution structures. This means that the structures in the current structure space define what is “normal” in a protein structure. The advantage of these analyses and derived parameters is that they do not take into account the process that leads to the model, only the final product and its reliability. The main disadvantage is that the current structure space is focused on proteins with known function and of biomedical or biotechnological interest.</p>
<p>One of the most common and powerful methods for assessing the stereochemistry of a protein is the <a href="intro.html#sec-rama">Ramachandran plot</a>, which was defined in 1963 and is still in use.</p>
<p>Another widely used analysis (available for all PDB structures) is the side <a href="https://www.wwpdb.org/validation/XrayValidationReportHelp#torsion_angles">chain torsion angles</a>, usually measured as <em><strong>Side chain outliers</strong>.</em> As described in the <a href="intro.html#sec-str">Introduction</a>, the amino acid side chains also have some preferred conformations. Like the Ramachandran plot, the plot of the χ1-χ2 torsion angles can indicate problems with a protein model if the angle values are outside of the high density values.</p>
<p>Bad contact or <a href="https://www.wwpdb.org/validation/XrayValidationReportHelp#close_contacts">clashes</a> indicate a poor model. It is obvious that two atoms cannot be in the same (or a very close) location. We can define this as a situation where two unbonded atoms have a center-to-center distance smaller than the sum of their van der Walls radii.</p>
</section>
</section>
<section id="sec-apps" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Protein structure display</h1>
<section id="protein-structure-file-formats" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="protein-structure-file-formats"><span class="header-section-number">4.1</span> Protein structure file formats</h2>
<p>Experimental structural data from different methods are stored in different file formats. For instance, raw crystallographic data are usually stored as <code>*.ccp4</code> files, but Cryo-EM or X-ray density maps can be stored in <code>*.mrc</code> or <code>*.mtz</code> files. Other complex file formats, such as the Extensible Markup Language <code>*.xml</code>, provide a framework for structure complex information and documents like protein structures.</p>
<p>Along with the establishment of the Protein Data Bank, a simple and standardized format was developed. The <em>Brookhaven</em> or <code>PDB</code> format consists of line records in a fixed format describing atomic coordinates, chemical and biochemical features, experimental details of the structure determination, and some structural features such as secondary structure assignments, hydrogen bonding, or active sites. The current version is named PDBx/mmCIF) also incorporates the expanded crystallographic information file format (mmCIF), which allows the representation of large structures, complex chemistry, and new and hybrid experimental methods. Thus a <code>*.pdb</code> and <code>*.cif</code> files can be considered as identical files.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
PDB-101
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Check PDB-101 course about PDBx/mmCIF format at PDB RCSB site <a href="https://pdb101.rcsb.org/learn/guide-to-understanding-pdb-data/beginner%E2%80%99s-guide-to-pdb-structures-and-the-pdbx-mmcif-format">here</a>.</p>
</div>
</div>
</div>
<div id="fig-PDBfile" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="images/paste-96FF761D.png" class="figure img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;8: Coordinates in the PDB file (6KI3)</figcaption>
</figure>
</div>
<section id="occupancy-and-b-factor" class="level3">
<h3 class="anchored" data-anchor-id="occupancy-and-b-factor">Occupancy and B-factor</h3>
<p>Except for the repetition of the atom type in the rightmost column, the last columns in the PDB file are the <strong>Occupancy</strong> and the <strong>temperature factor</strong> or the <strong>B-factor</strong>.</p>
<p>Macromolecular crystals consist of many individual molecules packed in a symmetrical arrangement. In some crystals there are slight differences between the individua molecules. For instance, a sidechain on the surface may wag back and forth between several conformations, or a substrate may bind in two orientations at an active site, or a metal ion may be detected as bound to only a few of the molecules. When researchers build the atomic model of these portions, they can use the occupancy to estimate the amount of each conformation observed in the crystal. Therefore, by definition, the sum of <strong>occupancy</strong> values for each atom must be 1. Usually, we see a single record for an atom, with an occupancy value of 1, indicating that the atom is found in all of the molecules in the same place in the crystal. However, if a metal ion binds to only half of the molecules in the crystal, the researcher sees a faint image of the ion in the electron density map and can assign an occupancy of 0.5 for this atom in the PDB structure file. For each atom, two (or more) atom records are included with occupancies such as 0.5 and 0.5, or 0.4 and 0.6, or other fractions of occupancies that sum to a total of 1.</p>
<p>On the other hand, the <strong>temperature value or <em>B</em>-factor is</strong> a measure of our confidence in the location of individual atoms, as described above (<a href="#sec-assess">Section&nbsp;3</a>). If you find an atom with a high temperature factor on the surface of a protein, keep in mind that this atom is likely to be moving around a lot and that the coordinates given in the PDB file are only a possible snapshot of its location. Thus, an atom dataset with an occupancy &lt; 1 may have a low B-factor if that position is safe.</p>
<p>As you can imagine, this column is also used by computationally derived models to indicate a confidence value that can be parsed for diverse purposes, including structure coloring.</p>
</section>
</section>
<section id="biological-macromolecules-display-applications" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="biological-macromolecules-display-applications"><span class="header-section-number">4.2</span> Biological macromolecules display applications</h2>
<section id="pymol" class="level3">
<h3 class="anchored" data-anchor-id="pymol">PyMOL</h3>
<p><a href="https://en.wikipedia.org/wiki/PyMOL">PyMOL</a> is a very powerful molecular visualization system written originally by <a href="https://en.wikipedia.org/wiki/Warren_Lyford_DeLano">Warren DeLano</a>. It was released in 2000 and soon became very popular. It’s currently commercialized under License by <a href="https://pymol.org/">Schrödinger</a> but a free license for teaching can be requested. Also, open source code is available on <a href="https://github.com/schrodinger/pymol-open-source">GitHub</a> that can be installed on Linux or MAC. More info on <a href="https://en.wikipedia.org/wiki/PyMOL">Wikipedia</a>. You can also check this quick <a href="https://www.uml.edu/docs/PyMOL%20Quick%20Reference%20Guide_tcm18-230352.pdf">Reference guide</a></p>
<p>PyMOL allows working with different structures representation, but also with raw experimental data in different <a href="https://pymol.org/dokuwiki/doku.php?id=format">formats</a>.</p>
<p>PyMOL is written in Python and can be used with interactive menus and also with command line. There are a lot of resources that can help you with PyMOL, like a <a href="https://pymol.org/dokuwiki/">Documentation Reference Wiki</a> or a community-supported <a href="https://pymolwiki.org/index.php/Main_Page">PyMOLWiki</a>. Moreover, it allows the implementation of new functionalities as plugins, like <a href="http://schubert.bio.uniroma1.it/pymod/index.html">PyMod</a> or <a href="http://schubert.bio.uniroma1.it/dockingpie/index.html">DockingPie</a>, among others. <a href="https://pymolwiki.org/index.php/PyMod">PyMod</a> <span class="citation" data-cites="janson2021">(<a href="#ref-janson2021" role="doc-biblioref">Janson and Paiardini 2021</a>)</span> is designed to act as simple and intuitive interface between PyMOL and several bioinformatics tools (i.e., PSI-BLAST, Clustal Omega, HMMER, MUSCLE, CAMPO, PSIPRED, and MODELLER). Starting from the amino acid sequence of the target protein, PyMod is designed to carry out the main steps of the homology modeling process (that is, template searching, target-template sequence alignment and model building) in order to build a 3D atomic model of a target protein (or protein complex). The integration with PyMOL facilitates a detailed analysis of the modeling process.</p>
<p>Finally, as any Python-based program, it can be used within Jupyter notebooks (see <a href="https://www.computer.org/csdl/magazine/cs/2021/02/09354947/1rgCkrAJCko" class="uri">https://www.computer.org/csdl/magazine/cs/2021/02/09354947/1rgCkrAJCko</a>).</p>
</section>
<section id="ucsf-chimerax" class="level3">
<h3 class="anchored" data-anchor-id="ucsf-chimerax">UCSF ChimeraX</h3>
<p><a href="https://www.rbvi.ucsf.edu/chimerax/">ChimeraX</a> <span class="citation" data-cites="pettersen2021">(<a href="#ref-pettersen2021" role="doc-biblioref">Pettersen et al. 2021</a>)</span> is a fully open source software, developed by the UCSF as a renovated version of the former <a href="https://www.cgl.ucsf.edu/chimera/">Chimera</a> software, with versions for Linux, MacOS, and Windows. It aims to be a comprehensive structural biology tool, but it is more widely known for its capacities for EM maps. As any other open source software, it has gained new and exciting capacities in the last years, like <a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/vr.html">Virtual Reality capabilities</a> or <a href="https://www.rbvi.ucsf.edu/chimerax/data/alphafold-nov2021/af_sbgrid.html">Alphafold2</a> modeling.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>There is an excellent ChimeraX User Guide, with examples at the RBVI@UCSF site <a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/index.html">here</a>.</p>
</div>
</div>
</section>
<section id="molecular-structures-on-your-website-litemol-and-others" class="level3">
<h3 class="anchored" data-anchor-id="molecular-structures-on-your-website-litemol-and-others">Molecular structures on your website: LiteMol and others</h3>
<p><a href="https://www.litemol.org/Viewer">LiteMol Viewer</a> is a powerful HTML5 web application for 3D visualization of molecules and other related data. It is used in a web browser, eliminating the need for external software and also allowing the integration with third-party sites as an embedded plugin. More information about LiteMol can be found on <span class="citation" data-cites="sehnal2017">Sehnal et al. (<a href="#ref-sehnal2017" role="doc-biblioref">2017</a>)</span>, the <a href="https://webchem.ncbr.muni.cz/Wiki/LiteMol:UserManual">wiki</a>, or <a href="https://www.youtube.com/channel/UCRoyYUeP1hdH2r8XUW-WMoA">YouTube tutorials</a>.</p>
<p>The same philosophy applies to other open-source viewers that were developed later and are now more widely used, like <a href="https://nglviewer.org/">NGL Viewer</a> and <a href="https://molstar.org/">Mol*</a> <span class="citation" data-cites="sehnal2021">Sehnal et al. (<a href="#ref-sehnal2021" role="doc-biblioref">2021</a>)</span>, used in <a href="https://www.rcsb.org/">RCSB-PDB</a> and <a href="https://www.ebi.ac.uk/pdbe/">PDBe</a> sites for 3D visualization of structures. With <em>Mol*</em> you can save your work session in <code>molj</code> (without the actual structures) or <code>molx</code> (with embedded structures) formats, as in the <a href="#fig-molstar">Figure 8</a> above.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Mol* &amp; Quarto
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Mol* can be very easily integrated in other third party services and in your own website. For instance, it has a <a href="https://github.com/jmbuhr/quarto-molstar">Quarto extension</a>, which prompted me to use it on this site.</p>
</div>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-6-contents" aria-controls="callout-6" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
A tribute to the pioneers
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-6" class="callout-6-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Other applications that you may know, hear about or came into but are now discontinued are:</p>
<ul>
<li><p><strong>SwissPDBViewer</strong> (aka DeepView), developed to work with SWISS-MODEL homology modeling app, is an application that provides a user-friendly interface allowing to analyze several proteins at the same time. It has currently fallen in disuse as the last version (4.1) is only a 32 bits application.</p></li>
<li><p><strong>RasMol</strong> and <strong>OpenRasMol</strong> were developed initially in 1992 and its last release was in 2009. It was a pioneer as a simple molecular display open-source application, but it is outdated nowadays.</p></li>
</ul>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="pymol-practice" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> <span style="color:green"><strong>PyMOL Practice</strong></span></h1>
<p>Our PyMOL Practice has two parts.</p>
<section id="part-a-a-10-steps-self-guided-practice" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="part-a-a-10-steps-self-guided-practice"><span class="header-section-number">5.1</span> <a href="https://www.evernote.com/shard/s62/sh/666c39a1-ac03-64c9-b8a0-cab9ff3959c7/9d20b6eadae8d130fdbe6652d60237f2">PART A: A 10-steps self-guided practice</a></h2>
<p>This is a <a href="https://evernote.com/intl/es">Evernote</a> note that you can consult online and also copy into your Evenote account if you wish.</p>
</section>
<section id="part-b-pymol-challenge" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="part-b-pymol-challenge"><span class="header-section-number">5.2</span> PART B: PyMOL Challenge</h2>
<p>Make a ready-to-publish picture of your favorite protein. As a suggestion, you can reproduce the top panels in the Figure 1B of <span class="citation" data-cites="gao2020">Gao et al. (<a href="#ref-gao2020" role="doc-biblioref">2020</a>)</span>, but any structure involving more than one domain and/or with a substrate/cofactor molecule can be a good challenge.</p>



</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-callaway2020" class="csl-entry" role="listitem">
Callaway, Ewen. 2020. <span>“Revolutionary Cryo-EM Is Taking over Structural Biology.”</span> <em>Nature</em> 578 (7794): 201–1. <a href="https://doi.org/10.1038/d41586-020-00341-9">https://doi.org/10.1038/d41586-020-00341-9</a>.
</div>
<div id="ref-egelman2016" class="csl-entry" role="listitem">
Egelman, Edward H. 2016. <span>“The Current Revolution in Cryo-EM.”</span> <em>Biophysical Journal</em> 110 (5): 1008–12. <a href="https://doi.org/10.1016/j.bpj.2016.02.001">https://doi.org/10.1016/j.bpj.2016.02.001</a>.
</div>
<div id="ref-gao2020" class="csl-entry" role="listitem">
Gao, Yan, Liming Yan, Yucen Huang, Fengjiang Liu, Yao Zhao, Lin Cao, Tao Wang, et al. 2020. <span>“Structure of the RNA-dependent RNA polymerase from COVID-19 virus.”</span> <em>Science (New York, N.Y.)</em> 368 (6492): 779–82. <a href="https://doi.org/10.1126/science.abb7498">https://doi.org/10.1126/science.abb7498</a>.
</div>
<div id="ref-structur" class="csl-entry" role="listitem">
Gu, J., and P. E. Bourne. 2009. <em>Structural Bioinformatics, 2nd Edition | Wiley</em>. <a href="https://www.wiley.com/en-us/Structural+Bioinformatics%2C+2nd+Edition-p-9780470181058">https://www.wiley.com/en-us/Structural+Bioinformatics%2C+2nd+Edition-p-9780470181058</a>.
</div>
<div id="ref-janson2021" class="csl-entry" role="listitem">
Janson, G., and A. Paiardini. 2021. <span>“PyMod 3: a complete suite for structural bioinformatics in PyMOL.”</span> <em>Bioinformatics</em> 37 (May): 1471–72. <a href="https://doi.org/10.1093/bioinformatics/btaa849">https://doi.org/10.1093/bioinformatics/btaa849</a>.
</div>
<div id="ref-pettersen2021" class="csl-entry" role="listitem">
Pettersen, Eric F., Thomas D. Goddard, Conrad C. Huang, Elaine C. Meng, Gregory S. Couch, Tristan I. Croll, John H. Morris, and Thomas E. Ferrin. 2021. <span>“UCSF ChimeraX: Structure visualization for researchers, educators, and developers.”</span> <em>Protein Science: A Publication of the Protein Society</em> 30 (1): 70–82. <a href="https://doi.org/10.1002/pro.3943">https://doi.org/10.1002/pro.3943</a>.
</div>
<div id="ref-sehnal2021" class="csl-entry" role="listitem">
Sehnal, David, Sebastian Bittrich, Mandar Deshpande, Radka Svobodová, Karel Berka, Václav Bazgier, Sameer Velankar, Stephen K. Burley, Jaroslav Koča, and Alexander S. Rose. 2021. <span>“Mol* Viewer: modern web app for 3D visualization and analysis of large biomolecular structures.”</span> <em>Nucleic Acids Research</em> 49 (W1): W431–37. <a href="https://doi.org/10.1093/nar/gkab314">https://doi.org/10.1093/nar/gkab314</a>.
</div>
<div id="ref-sehnal2017" class="csl-entry" role="listitem">
Sehnal, David, Mandar Deshpande, Radka Svobodová Vařeková, Saqib Mir, Karel Berka, Adam Midlik, Lukáš Pravda, Sameer Velankar, and Jaroslav Koča. 2017. <span>“LiteMol suite: interactive web-based visualization of large-scale macromolecular structure data.”</span> <em>Nature Methods</em> 14 (12): 1121–22. <a href="https://doi.org/10.1038/nmeth.4499">https://doi.org/10.1038/nmeth.4499</a>.
</div>
</div></section></div></main> <!-- /main -->
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  var t = document.getElementsByClassName("webex-total_correct");
  for (var i = 0; i < t.length; i++) {
    p = t[i].parentElement;
    var correct = p.getElementsByClassName("webex-correct").length;
    var solvemes = p.getElementsByClassName("webex-solveme").length;
    var radiogroups = p.getElementsByClassName("webex-radiogroup").length;
    var selects = p.getElementsByClassName("webex-select").length;

    t[i].innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");

  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* check answers */
check_func = function() {
  console.log("webex: check answers");

  var cl = this.parentElement.classList;
  if (cl.contains('unchecked')) {
    cl.remove("unchecked");
    this.innerHTML = "Hide Answers";
  } else {
    cl.add("unchecked");
    this.innerHTML = "Show Answers";
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");

  var cl = this.classList

  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;

  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }

  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }

  update_total_correct();
}

window.onload = function() {
  console.log("webex onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  var check_sections = document.getElementsByClassName("webex-check");
  console.log("check:", check_sections.length);
  for (var i = 0; i < check_sections.length; i++) {
    check_sections[i].classList.add("unchecked");

    let btn = document.createElement("button");
    btn.innerHTML = "Show Answers";
    btn.classList.add("webex-check-button");
    btn.onclick = check_func;
    check_sections[i].appendChild(btn);

    let spn = document.createElement("span");
    spn.classList.add("webex-total_correct");
    check_sections[i].appendChild(spn);
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;

    solveme[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }

  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
    selects[i].insertAdjacentHTML("afterend", " <span class='webex-icon'></span>")
  }

  update_total_correct();
}

</script>
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
<nav class="page-navigation column-body">
  <div class="nav-page nav-page-previous">
      <a href="./ddbb.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Protein Databases</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./homology.html" class="pagination-link">
        <span class="nav-page-text">Homology Modeling</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">Modesto Redrejo Rodríguez, 2023</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://creativecommons.org/licenses/by-nc/2.0/">
      <i class="bi bi-badge-cc-fill" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://twitter.com/mredrejo"><iconify-icon inline="" icon="fa6-brands:x-twitter"></iconify-icon></a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link active" href="https://github.com/StrBio/strbio.github.io" aria-current="page">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



<script src="site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>